<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Design target specific gRNAs using CRISPRseek package • RISPRseekGUIDEseqBioc2020Workshop</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Design target specific gRNAs using CRISPRseek package">
<meta property="og:description" content="RISPRseekGUIDEseqBioc2020Workshop">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">RISPRseekGUIDEseqBioc2020Workshop</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/CRISPRdemo.html">Design target specific gRNAs using CRISPRseek package</a>
    </li>
    <li>
      <a href="../articles/GUIDEseqdemo.html">Analysis of GUIDE-seq data to identify offtargets using GUIDEseq package</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/LihuaJulieZhu/CRISPERseekGUIDERseqBioc2020Workshop">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="CRISPRdemo_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Design target specific gRNAs using CRISPRseek package</h1>
                        <h4 class="author">Lihua Julie Zhu</h4>
            
            <h4 class="date">2020-07-13</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/LihuaJulieZhu/CRISPRseekGUIDEseqBioc2020Workshop/blob/master/vignettes/CRISPRdemo.Rmd"><code>vignettes/CRISPRdemo.Rmd</code></a></small>
      <div class="hidden name"><code>CRISPRdemo.Rmd</code></div>

    </div>

    
    
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<p>Zhu LJ, Holmes BR, Aronin N and Brodsky MH (2014). “CRISPRseek: A Bioconductor Package to Identify Target-Specific Guide RNAs for CRISPR-Cas9 Genome-Editing Systems.” PLoS one, 9(9). <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4172692/" class="uri">http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4172692/</a>.</p>
<p>Zhu LJ (2015). “Overview of guide RNA design tools for CRISPR-Cas9 genome editing technology.” Front. Biol., 10(4).</p>
</div>
<div id="first-load-the-required-packages-and-specify-the-input-file-path-" class="section level1">
<h1 class="hasAnchor">
<a href="#first-load-the-required-packages-and-specify-the-input-file-path-" class="anchor"></a>First load the required packages and specify the input file path.</h1>
<p>We are going to use a sequence from human as input, which has been included as as fasta file in the CRISPRseek package.<br>
To perform off-target analysis, we need to load Human BSgenome package.<br>
To annotate the target and off-targets, we need to load Human Transcript and gene identifier mapping packages.<br>
In addition, you need to specify the output directory which will be the directory to look for all the output files.<br>
For the current release, you no longer need to specify the file containing all restriction enzyme (RE) cut patterns. You have the option to specify your own RE pattern file instead of the default one supplied by the CRISPR package.</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">CRISPRseek</span>)</pre></body></html></div>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## Loading required package: parallel</code></pre>
<pre><code>## 
## Attaching package: 'BiocGenerics'</code></pre>
<pre><code>## The following objects are masked from 'package:parallel':
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
##     union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>## Loading required package: Biostrings</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## 
## Attaching package: 'S4Vectors'</code></pre>
<pre><code>## The following object is masked from 'package:base':
## 
##     expand.grid</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## Loading required package: XVector</code></pre>
<pre><code>## 
## Attaching package: 'Biostrings'</code></pre>
<pre><code>## The following object is masked from 'package:base':
## 
##     strsplit</code></pre>
<div class="sourceCode" id="cb17"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">BSgenome.Hsapiens.UCSC.hg19</span>)</pre></body></html></div>
<pre><code>## Loading required package: BSgenome</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<pre><code>## Loading required package: GenomicRanges</code></pre>
<pre><code>## Loading required package: rtracklayer</code></pre>
<div class="sourceCode" id="cb22"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">TxDb.Hsapiens.UCSC.hg19.knownGene</span>)</pre></body></html></div>
<pre><code>## Loading required package: GenomicFeatures</code></pre>
<pre><code>## Loading required package: AnnotationDbi</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation("Biobase")', and for packages 'citation("pkgname")'.</code></pre>
<div class="sourceCode" id="cb27"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">org.Hs.eg.db</span>)</pre></body></html></div>
<pre><code>## </code></pre>
<div class="sourceCode" id="cb29"><html><body><pre class="r"><span class="no">outputDir</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span>(),<span class="st">"CRISPRseekDemo"</span>)
<span class="no">inputFilePath</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">'extdata'</span>, <span class="st">'inputseq.fa'</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">'CRISPRseek'</span>)</pre></body></html></div>
</div>
<div id="here-is-the-command-to-learn-more-about-the-crisprseek-package-and-how-to-use-the-two-workflow-functions-to-find-grnas-and-perform-off-target-analysis-with-different-use-cases-" class="section level1">
<h1 class="hasAnchor">
<a href="#here-is-the-command-to-learn-more-about-the-crisprseek-package-and-how-to-use-the-two-workflow-functions-to-find-grnas-and-perform-off-target-analysis-with-different-use-cases-" class="anchor"></a>Here is the command to learn more about the CRISPRseek package and how to use the two workflow functions to find gRNAs and perform off-target analysis with different use cases.</h1>
<div class="sourceCode" id="cb30"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/args.html">args</a></span>(<span class="no">offTargetAnalysis</span>)
<span class="fu"><a href="https://rdrr.io/r/base/args.html">args</a></span>(<span class="no">compare2Sequences</span>)
?<span class="no">offTargetAnalysis</span>
?<span class="no">compare2Sequences</span>
?<span class="no">CRISPRseek</span>
<span class="fu"><a href="https://rdrr.io/r/utils/browseVignettes.html">browseVignettes</a></span>(<span class="st">'CRISPRseek'</span>)</pre></body></html></div>
</div>
<div id="scenario-1-find-all-grnas-and-perform-off-target-analysis-for-each-grna" class="section level1">
<h1 class="hasAnchor">
<a href="#scenario-1-find-all-grnas-and-perform-off-target-analysis-for-each-grna" class="anchor"></a>Scenario 1: Find all gRNAs and perform off-target analysis for each gRNA</h1>
<p>Please note that chromToSearch is set to chrX here for speed purpose, usually you do not need to set it, by default it is set to all.</p>
<div class="sourceCode" id="cb31"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/CRISPRseek/man/offTargetAnalysis.html">offTargetAnalysis</a></span>(<span class="no">inputFilePath</span>,
    <span class="kw">BSgenomeName</span> <span class="kw">=</span> <span class="no">Hsapiens</span>,
    <span class="kw">chromToSearch</span> <span class="kw">=</span><span class="st">"chrX"</span>,
    <span class="kw">txdb</span> <span class="kw">=</span> <span class="no">TxDb.Hsapiens.UCSC.hg19.knownGene</span>,
    <span class="kw">orgAnn</span> <span class="kw">=</span> <span class="no">org.Hs.egSYMBOL</span>,
    <span class="kw">outputDir</span> <span class="kw">=</span> <span class="no">outputDir</span>,
    <span class="kw">overwrite</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<pre><code>## Validating input ...
## Searching for gRNAs ...
## &gt;&gt;&gt; Finding all hits in sequence chrX ...
## &gt;&gt;&gt; DONE searching
## Building feature vectors for scoring ...
## Calculating scores ...
## Annotating, filtering and generating reports ...</code></pre>
<pre><code>##   403 genes were dropped because they have exons located on both strands
##   of the same reference sequence or on more than one reference sequence,
##   so cannot be represented by a single genomic range.
##   Use 'single.strand.genes.only=FALSE' to get all the genes in a
##   GRangesList object, or use suppressMessages() to suppress this message.</code></pre>
<pre><code>## Done annotating
## Add paired information...
## Add RE information...
## write gRNAs to bed file...
## Scan for REsites in flanking region...
## Done. Please check output files in directory  /__w/CRISPRseekGUIDEseqBioc2020Workshop/CRISPRseekGUIDEseqBioc2020Workshop/vignettes/CRISPRseekDemo/</code></pre>
<pre><code>## $on.target
##      names                  forViewInUCSC           
## [1,] "Hsap_GATA1_ex2_gR17f" "chrX:48649564-48649586"
## [2,] "Hsap_GATA1_ex2_gR20r" "chrX:48649577-48649599"
## [3,] "Hsap_GATA1_ex2_gR39f" "chrX:48649586-48649608"
## [4,] "Hsap_GATA1_ex2_gR40f" "chrX:48649587-48649609"
## [5,] "Hsap_GATA1_ex2_gR7r"  "chrX:48649564-48649586"
##      extendedSequence                 gRNAefficacy        
## [1,] "TCCCCCAGTTTGTGGATCCTGCTCTGGTGT" "0.106547942216103" 
## [2,] "TTCTGGTGTGGAGGACACCAGAGCAGGATC" "0.142091080678942" 
## [3,] "TCTGGTGTCCTCCACACCAGAATCAGGGGT" "0.0696295234374132"
## [4,] "CTGGTGTCCTCCACACCAGAATCAGGGGTT" "0.261881755628254" 
## [5,] "GACACCAGAGCAGGATCCACAAACTGGGGG" "0.101719829329578" 
## 
## $summary
##                  names          forViewInUCSC               extendedSequence
## 1 Hsap_GATA1_ex2_gR17f chrX:48649564-48649586 TCCCCCAGTTTGTGGATCCTGCTCTGGTGT
## 2 Hsap_GATA1_ex2_gR20r chrX:48649577-48649599 TTCTGGTGTGGAGGACACCAGAGCAGGATC
## 3 Hsap_GATA1_ex2_gR39f chrX:48649586-48649608 TCTGGTGTCCTCCACACCAGAATCAGGGGT
## 4 Hsap_GATA1_ex2_gR40f chrX:48649587-48649609 CTGGTGTCCTCCACACCAGAATCAGGGGTT
## 5  Hsap_GATA1_ex2_gR7r chrX:48649564-48649586 GACACCAGAGCAGGATCCACAAACTGGGGG
##         gRNAefficacy            gRNAsPlusPAM top5OfftargetTotalScore
## 1  0.106547942216103 CCAGTTTGTGGATCCTGCTCNGG                     3.3
## 2  0.142091080678942 GGTGTGGAGGACACCAGAGCNGG                     3.4
## 3 0.0696295234374132 GTGTCCTCCACACCAGAATCNGG                     1.1
## 4  0.261881755628254 TGTCCTCCACACCAGAATCANGG                     2.9
## 5  0.101719829329578 CCAGAGCAGGATCCACAAACNGG                     0.2
##   top10OfftargetTotalScore top1Hit.onTarget.MMdistance2PAM
## 1                      3.3                             NMM
## 2                      3.4                             NMM
## 3                      1.1                             NMM
## 4                      2.9                             NMM
## 5                      0.2                             NMM
##   topOfftarget1MMdistance2PAM topOfftarget2MMdistance2PAM
## 1                    20,17,13                     20,19,3
## 2                    18,16,15                     17,14,1
## 3                     16,15,1                     15,11,3
## 4                        13,2                            
## 5                      16,8,3                            
##   topOfftarget3MMdistance2PAM topOfftarget4MMdistance2PAM
## 1                                                        
## 2                     20,14,8                     13,10,7
## 3                                                        
## 4                                                        
## 5                                                        
##   topOfftarget5MMdistance2PAM topOfftarget6MMdistance2PAM
## 1                                                        
## 2                                                        
## 3                                                        
## 4                                                        
## 5                                                        
##   topOfftarget7MMdistance2PAM topOfftarget8MMdistance2PAM
## 1                                                        
## 2                                                        
## 3                                                        
## 4                                                        
## 5                                                        
##   topOfftarget9MMdistance2PAM topOfftarget10MMdistance2PAM          REname
## 1                                                                         
## 2                                                               Aco12261II
## 3                                                          BslI HinfI TfiI
## 4                                                          BslI HinfI TfiI
## 5                                                               BslI PflMI
##   uniqREin200 uniqREin100
## 1                        
## 2                        
## 3  HinfI TfiI  HinfI TfiI
## 4  HinfI TfiI  HinfI TfiI
## 5       PflMI       PflMI
## 
## $offtarget
##                    name             gRNAPlusPAM       OffTargetSequence inExon
## 1  Hsap_GATA1_ex2_gR17f CCAGTTTGTGGATCCTGCTCNGG ACAATTTCTGGATCCTGCTCCAG       
## 2  Hsap_GATA1_ex2_gR17f CCAGTTTGTGGATCCTGCTCNGG CCAGTTTGTGGATCCTGCTCTGG   TRUE
## 3  Hsap_GATA1_ex2_gR17f CCAGTTTGTGGATCCTGCTCNGG GTAGTTTGTGGATCCTGTTCTAG       
## 4  Hsap_GATA1_ex2_gR20r GGTGTGGAGGACACCAGAGCNGG GGAGCTGAGGACACCAGAGCGGG       
## 5  Hsap_GATA1_ex2_gR20r GGTGTGGAGGACACCAGAGCNGG GGTATGTAGGACACCAGAGACAG       
## 6  Hsap_GATA1_ex2_gR20r GGTGTGGAGGACACCAGAGCNGG GGTGTGGAGGACACCAGAGCAGG   TRUE
## 7  Hsap_GATA1_ex2_gR20r GGTGTGGAGGACACCAGAGCNGG GGTGTGGGGGTCAGCAGAGCCAG       
## 8  Hsap_GATA1_ex2_gR20r GGTGTGGAGGACACCAGAGCNGG TGTGTGTAGGACTCCAGAGCAAG       
## 9  Hsap_GATA1_ex2_gR39f GTGTCCTCCACACCAGAATCNGG GTGTCCTCCACACCAGAATCAGG   TRUE
## 10 Hsap_GATA1_ex2_gR39f GTGTCCTCCACACCAGAATCNGG GTGTCTTCCCCACCAGATTCTAG       
## 11 Hsap_GATA1_ex2_gR39f GTGTCCTCCACACCAGAATCNGG GTGTTTTCCACACCAGAATGCAG       
## 12 Hsap_GATA1_ex2_gR40f TGTCCTCCACACCAGAATCANGG TGTCCTCAACACCAGAATGATAG       
## 13 Hsap_GATA1_ex2_gR40f TGTCCTCCACACCAGAATCANGG TGTCCTCCACACCAGAATCAGGG   TRUE
## 14  Hsap_GATA1_ex2_gR7r CCAGAGCAGGATCCACAAACNGG CCAGAGCAGGATCCACAAACTGG   TRUE
## 15  Hsap_GATA1_ex2_gR7r CCAGAGCAGGATCCACAAACNGG CCAGTGCAGGATTCACATACTAG       
##    inIntron entrez_id    gene score n.mismatch mismatch.distance2PAM
## 1      TRUE       618  BCYRN1   2.6          3              20,17,13
## 2                2623   GATA1   100          0                      
## 3      TRUE      1183   CLCN4   0.7          3               20,19,3
## 4                               1.4          3              18,16,15
## 5                                 1          3               17,14,1
## 6                2623   GATA1   100          0                      
## 7                               0.2          3               13,10,7
## 8      TRUE     57477 SHROOM4   0.8          3               20,14,8
## 9                2623   GATA1   100          0                      
## 10     TRUE     10149  ADGRG2   0.3          3               15,11,3
## 11                              0.8          3               16,15,1
## 12     TRUE    139324     HDX   2.9          2                  13,2
## 13               2623   GATA1   100          0                      
## 14               2623   GATA1   100          0                      
## 15                              0.2          3                16,8,3
##               alignment isCanonicalPAM            forViewInUCSC strand chrom
## 1  A..A...C............              0   chrX:70453482-70453504      -  chrX
## 2  ....................              1   chrX:48649564-48649586      +  chrX
## 3  GT...............T..              0   chrX:10161855-10161877      +  chrX
## 4  ..A.CT..............              1 chrX:152513913-152513935      -  chrX
## 5  ...A..T............A              0   chrX:14446436-14446458      -  chrX
## 6  ....................              1   chrX:48649577-48649599      -  chrX
## 7  .......G..T..G......              0   chrX:50699578-50699600      +  chrX
## 8  T.....T.....T.......              0   chrX:50383047-50383069      -  chrX
## 9  ....................              1   chrX:48649586-48649608      +  chrX
## 10 .....T...C.......T..              0   chrX:19043357-19043379      +  chrX
## 11 ....TT.............G              0   chrX:81194638-81194660      +  chrX
## 12 .......A..........G.              0   chrX:83738197-83738219      -  chrX
## 13 ....................              1   chrX:48649587-48649609      +  chrX
## 14 ....................              1   chrX:48649564-48649586      -  chrX
## 15 ....T.......T....T..              0   chrX:66676065-66676087      -  chrX
##    chromStart  chromEnd               extendedSequence gRNAefficacy
## 1    70453482  70453504                           &lt;NA&gt;           NA
## 2    48649564  48649586 TCCCCCAGTTTGTGGATCCTGCTCTGGTGT   0.10654794
## 3    10161855  10161877                           &lt;NA&gt;           NA
## 4   152513913 152513935                           &lt;NA&gt;           NA
## 5    14446436  14446458                           &lt;NA&gt;           NA
## 6    48649577  48649599 TTCTGGTGTGGAGGACACCAGAGCAGGATC   0.14209108
## 7    50699578  50699600                           &lt;NA&gt;           NA
## 8    50383047  50383069                           &lt;NA&gt;           NA
## 9    48649586  48649608 TCTGGTGTCCTCCACACCAGAATCAGGGGT   0.06962952
## 10   19043357  19043379                           &lt;NA&gt;           NA
## 11   81194638  81194660                           &lt;NA&gt;           NA
## 12   83738197  83738219                           &lt;NA&gt;           NA
## 13   48649587  48649609 CTGGTGTCCTCCACACCAGAATCAGGGGTT   0.26188176
## 14   48649564  48649586 GACACCAGAGCAGGATCCACAAACTGGGGG   0.10171983
## 15   66676065  66676087                           &lt;NA&gt;           NA
##                                                                                                                                                                                                                                                                                                                                                                                                                              flankSequence
## 1  GCTGGAGCAGAAACAGAAAGGGGTTAGGGGAGACTGTGGGAAAATAAAGTGTGTTTGGGATCCCTAAGGTGAAGGGAGAGAAAGGGTGACAGTTTCAACAGCTTATCTTGACTTAGGCCAAAGAAGGTTAAGGGGTTCTGAATAACTGTCTCTCCACATCTTCCCTGCACTCCCCAGGTTGCCTCCCTTCCCAGGGATGGACAATTTCTGGATCCTGCTCCAGGTAGACATCCTGGGAAGGGCCCTCTCCCCACAAAACTGCCCTTCCCTGCCTGAGTGCAGGGCAGCCCCTTCCTCAACTTTCACCATTTCTCTCTCTTAACCTTAGGGATCCTGTACAAACATGTTGTTTTCCAGCACATTAGTACTTAGGGATATTTTCATATCATACTGAGCTCCCCACCCCCACCCCACCCAACATAT
## 2  ATGGGGAGGTGGGAAGGAGAAATATGGAGACTGAGGTGATGGAGTGGGAGGAGGGGGAAAGGAGGAAGAGGAGCAGGTGAAAGGATGTGGGGTGGAAGGATTTCTGTGTCTGAGGACCCCTTCTGTCCTCGCAGGTTAATCCCCAGAGGCTCCATGGAGTTCCCTGGCCTGGGGTCCCTGGGGACCTCAGAGCCCCTCCCCCAGTTTGTGGATCCTGCTCTGGTGTCCTCCACACCAGAATCAGGGGTTTTCTTCCCCTCTGGGCCTGAGGGCTTGGATGCAGCAGCTTCCTCCACTGCCCCGAGCACAGCCACCGCTGCAGCTGCGGCACTGGCCTACTACAGGGACGCTGAGGCCTACAGACACTCCCCAGGTAACTCCATTGAGTGGCTGTCTTGGCATTGGCTGAGTGCTGTTGGGGTT
## 3  GTCAGCAAACTTTTTCTGCAAGAGGCCAGAGAGTAAATATTTTCAGGTTTGTGGCCCATGTGGTTTCTATCTCAGCTCTTCAGTTCTGCCACTGTAGCATGAAGGCAGCCACAGAGTGCATGTAATTGAATGGTGTAGCTGTGCTCCAATAAAACTTTATTTACAAAAACAGGCTGTGGGCCAGATTTGGCCCATGGGCTGTAGTTTGTGGATCCTGTTCTAGAGGAATGAGCTAAGGCCATTGCCTTAGCTGCATGGGAATCTCCTGGTAGATTGAGGGGATGGACTGGGCCAAGGGAAGGAAAGAGATTGGATGGTGACTAGAGGGAAAGTGGAAACAGGGAGGGTTTTTTTTTTTTAATGTTTAAATGCTGATGGCAAGGAGAGGAAGAGGTTGAAGATACAGGAGAAGGAAAGGGGAAT
## 4  GCCTTGACTGTACCATGGTCACACATTTCTTATCCAAAAGCAAAGGGATTGTTCATATAAAGAAATTATGAGTGTTGGTTGATCAGCTCAGTTTTGGAAATGTTAAAGTTGAGGTGCATATGAAGCACCTGAGGGCGGATGTCAAGTGGGGATAAGCAATATACCCTGGATATATGAGCCTCATGTTCAGGAGAGTGGTAGGAGCTGAGGACACCAGAGCGGGAATTAATATCACACAGATTGCGTTTAATCAAGGAGACTAGGAGAGATCACCTAGGGAATGACTGTAAACAAAAAAGGGAAGTCTAAAGACTGATTCCGCGTTCCTTCAATACTTTGAGCTACCGCAGAATTGGGGCATCAGTAATGAAGCCGAAGAATGAGTGAACAAATAAGTAAAAGAAAACCAGCTGAAACACAGTA
## 5  ACATTTTCCCCATGACATTCTTTTAATTGCAAATGCTCCTGAGAAGAGGAATAACACATGAAACGAATCCGCTCTGATCACAGATTTTTCTCCTTCTCAGATAAATCCAATTAACAGAAGATGTCTTAGAATTTAAGGGGGGGCAGAAAGAGAATCTCTGTGGGTAGAGGGTTGGGGAAAAAGCATGGGATGGTAGTTTTGGTATGTAGGACACCAGAGACAGTCCTCACCTAATTCAATGGCTTAAAATTATACCAAAGGAAAACAACTCATAACCCAAAAGGAATTAGGATGCACACACCAGGCTTACTCAGACATGTGGTGCTGCCTCCTATTTTACTTTATGGTAGATAAAGTTGAGCAGTTGTATAAATAGAAATGACTGGAGCCTTTACGTCTAATAGCAGAACTCATTTTTTATTT
## 6  GGATCTCCATGGCAACCCCAACAGCACTCAGCCAATGCCAAGACAGCCACTCAATGGAGTTACCTGGGGAGTGTCTGTAGGCCTCAGCGTCCCTGTAGTAGGCCAGTGCCGCAGCTGCAGCGGTGGCTGTGCTCGGGGCAGTGGAGGAAGCTGCTGCATCCAAGCCCTCAGGCCCAGAGGGGAAGAAAACCCCTGATTCTGGTGTGGAGGACACCAGAGCAGGATCCACAAACTGGGGGAGGGGCTCTGAGGTCCCCAGGGACCCCAGGCCAGGGAACTCCATGGAGCCTCTGGGGATTAACCTGCGAGGACAGAAGGGGTCCTCAGACACAGAAATCCTTCCACCCCACATCCTTTCACCTGCTCCTCTTCCTCCTTTCCCCCTCCTCCCACTCCATCACCTCAGTCTCCATATTTCTCCTT
## 7  AGACATAAGATGAAAATAAAGATTTTAAGAGAGAGACCATGAGTGGTGGGCCAATACCTTTATTGGTCCAGGGCATTATCCAAACAGGTTTCCCTCAGGGAGTTTTAACTGGTGGGTTTAAAGCAATCAGGCATGAATTCCAATAGGTAACGCTGTGTCTGAGAGGTGGTCACTGTGGCATATCTGCACAGTGCATGCAGGGTGTGGGGGTCAGCAGAGCCAGTCAAGAGGTTGTATCTAGCTATCTTATAGAGAAGTGATCACCAAGAGGAAGTTGTCTAAGACACATATCTGGATCAACCACATTATGAAACTGGGGGTGGGGAGGGTGTAGAACTGGAAACTGTGGCCAGGCGCGGTGGCTCATGCCTGTAATCCCAGCACTTTGGGAGGCCGAGGTGGACAGATCACGAGGTCAGGAGT
## 8  TCAAGAACCCTAAATATGGAAAAAGGCTTGAAATAGGAAGGATACCCTGTTCACCACTATATCCTCAGTGTCTAAAAGCATAGGTCATAGTAGATACCCATGGTACATAGTACTACTACTGGTAGATGCTTGACAAATAGTTGTTGACCGGATAACTACCTGGTAGCAGCAGCTGGAGAGATTTCCCGGCAGTGTGGCTTTGTGTGTAGGACTCCAGAGCAAGCATAGCCCCCAAGGTCTCCTTGTGCCTTCAGCCCCCACTCTGGGCACTCTGAAATCAATAATAGAACAATGTGATTACAAAGCTAGGGGCTTCACCTGGCTTTCCTCCGCTTAAATCTTAGCCACTGTTGCTGACCTGTGGCTATACTTCCTGGCCTTTGACAAACAGCTTCCCCTGCCCTTCCTCCTATACCTCCTGGT
## 9  TATGGAGACTGAGGTGATGGAGTGGGAGGAGGGGGAAAGGAGGAAGAGGAGCAGGTGAAAGGATGTGGGGTGGAAGGATTTCTGTGTCTGAGGACCCCTTCTGTCCTCGCAGGTTAATCCCCAGAGGCTCCATGGAGTTCCCTGGCCTGGGGTCCCTGGGGACCTCAGAGCCCCTCCCCCAGTTTGTGGATCCTGCTCTGGTGTCCTCCACACCAGAATCAGGGGTTTTCTTCCCCTCTGGGCCTGAGGGCTTGGATGCAGCAGCTTCCTCCACTGCCCCGAGCACAGCCACCGCTGCAGCTGCGGCACTGGCCTACTACAGGGACGCTGAGGCCTACAGACACTCCCCAGGTAACTCCATTGAGTGGCTGTCTTGGCATTGGCTGAGTGCTGTTGGGGTTGCCATGGAGATCCTTGGCTAGG
## 10 TGCTGGGATTATAGTCATGAGCCACCATGCCCCACCTACAGGTCACCTTCTTTCTGGGGGTTCCCCTATCCCTCCGCTGCAGTCCGGTTTAATTGCTCCTGCTGCTCTACATTCAGAGACTTAGAACATCCATCCTTTGGTACCCTATACAAGCCACCTTCAGTGCAGGGGTTGCACTGTATTGTTATTTGTTGCATCTTGTGTCTTCCCCACCAGATTCTAGAGCCCTGGTTCTCACAATTAGCCTCATGTTGGAAACATGCCAATGCTGGGTCTCAGCCCCTAAATACTGATTTGTTATGGAGCATGCCTGGGCACTGGGATTTTTGAAAGCTCCCAGGTGATTTTAATATGCAACCAGGGATGAGAAACATTGCTCTAGACTTGGTTTCCTATCTCAGAGAGTTGGGTTTATCCCTGTGG
## 11 AAGGAAGTAATTAAGATTAAATGAGGCTTGGAGAATGGGGCTCTAATTCGATAGGATTAATATCATTATAAGAAGAGGTACCAGAGTGTTGGCATGTGCATGCTCTCTGTCTTTCTTCGTGTGTGTGTGTGTGTGTATGTGTGTGTTTCTCCATGTGTATGCACTGAGGAAAGGACATGTGAGGATATAGAAAGAAGGCAGTGTTTTCCACACCAGAATGCAGCCCTTATCAGAAGTCAAATAAACCAGAACCTTTATCTTAGACTTCTAGCCTCTAAAACACTGAAAAATAAATTTTCGTTTAAACTGCCAGGTCTATAGTACCTTGTTACAGCAGCCTAAGGTGACTAATATAATTGTCAAGGTAGTGTTACTGGGAATTTTGTATTGGGATGGATTTTTTAACTGAGGAGATCAACACAG
## 12 GCAGGTGCACATAAGTCAAGAATTGAGGTTTGGGAACCTCCACCTAGATTTCAGAGGATGTATGAAAATGTCTGGATGTCCCTGCAGACATTTGCTGCATGGGTGGAGCCCTCATGGAGAACCTCTGCTAGGGCAATGTGGAAGAGACATGTGGGGTTGGAGCCCCAACACAGATAGTGGAGCTGTGAGAAGAGGGCCACTGTCCTCAACACCAGAATGATAGATCCACCGACAGCTTGCACCATGAGCCTGGGAAAGCCACATACACTTAACATCAGTCTGTGAAATCAGCTGAGGGGGGCGCTGTACCCTGCAAAGCCACAAAGTTGGAGCTGCCCAAGCCCTTTGATGCCCACTCCTTGCATCAGCATAACCTGAATGTGGGACATGGAGTCAAAGGAGATCACTTTGGAACTTTAAGTT
## 13 ATGGAGACTGAGGTGATGGAGTGGGAGGAGGGGGAAAGGAGGAAGAGGAGCAGGTGAAAGGATGTGGGGTGGAAGGATTTCTGTGTCTGAGGACCCCTTCTGTCCTCGCAGGTTAATCCCCAGAGGCTCCATGGAGTTCCCTGGCCTGGGGTCCCTGGGGACCTCAGAGCCCCTCCCCCAGTTTGTGGATCCTGCTCTGGTGTCCTCCACACCAGAATCAGGGGTTTTCTTCCCCTCTGGGCCTGAGGGCTTGGATGCAGCAGCTTCCTCCACTGCCCCGAGCACAGCCACCGCTGCAGCTGCGGCACTGGCCTACTACAGGGACGCTGAGGCCTACAGACACTCCCCAGGTAACTCCATTGAGTGGCTGTCTTGGCATTGGCTGAGTGCTGTTGGGGTTGCCATGGAGATCCTTGGCTAGGT
## 14 AACCCCAACAGCACTCAGCCAATGCCAAGACAGCCACTCAATGGAGTTACCTGGGGAGTGTCTGTAGGCCTCAGCGTCCCTGTAGTAGGCCAGTGCCGCAGCTGCAGCGGTGGCTGTGCTCGGGGCAGTGGAGGAAGCTGCTGCATCCAAGCCCTCAGGCCCAGAGGGGAAGAAAACCCCTGATTCTGGTGTGGAGGACACCAGAGCAGGATCCACAAACTGGGGGAGGGGCTCTGAGGTCCCCAGGGACCCCAGGCCAGGGAACTCCATGGAGCCTCTGGGGATTAACCTGCGAGGACAGAAGGGGTCCTCAGACACAGAAATCCTTCCACCCCACATCCTTTCACCTGCTCCTCTTCCTCCTTTCCCCCTCCTCCCACTCCATCACCTCAGTCTCCATATTTCTCCTTCCCACCTCCCCAT
## 15 TTTGCACAGTTTGGTGGAAAAAAGAATGGTTTCCCAGCTGGGTAGCACACTCACTCACTGCCTACCTTGGCTAGAGGTTGGGGACTCCCCTGCCCTGTGTGGCTATCAGGTGGGCCACCGCAACACACTTCTCTTTTCTCTCCATGTACAACACCAGCCGCCTAGTCAATTCTGATAAGATAAACTAGATACCTTGGTTGCCAGTGCAGGATTCACATACTAGTTATTATTTATTTTTTATGGGAGCCTCTGATCCCCAGTGCTTTTGTCATCCATCTTGGCCCCTCCCCAATATTTTCTTCTAAGACTCTTATAGTTTTAAATCTAATGTTTAGGCTTTTGATCCATTTTAAGTTAATTTTGTATAAAGAAGGGGTCCAATTTCATACTTTTACATGTGGATATTCAGTTTTTCTAATACAA
## 
## $gRNAs.bedFormat
##      [,1]   [,2]       [,3]       [,4]                   [,5]              
## [1,] "chrX" "48649563" "48649586" "Hsap_GATA1_ex2_gR17f" "106.547942216103"
## [2,] "chrX" "48649576" "48649599" "Hsap_GATA1_ex2_gR20r" "142.091080678942"
## [3,] "chrX" "48649585" "48649608" "Hsap_GATA1_ex2_gR39f" "69.6295234374132"
## [4,] "chrX" "48649586" "48649609" "Hsap_GATA1_ex2_gR40f" "261.881755628254"
## [5,] "chrX" "48649563" "48649586" "Hsap_GATA1_ex2_gR7r"  "101.719829329578"
##      [,6] [,7]       [,8]      
## [1,] "+"  "48649579" "48649581"
## [2,] "-"  "48649581" "48649583"
## [3,] "+"  "48649601" "48649603"
## [4,] "+"  "48649602" "48649604"
## [5,] "-"  "48649568" "48649570"
## 
## $REcutDetails
##                gRNAPlusPAM        REcutgRNAName     REname   REpattern
## 2  GGTGTGGAGGACACCAGAGCAGG Hsap_GATA1_ex2_gR20r Aco12261II      CCRGAG
## 3  GTGTCCTCCACACCAGAATCAGG Hsap_GATA1_ex2_gR39f       BslI CCNNNNNNNGG
## 4  TGTCCTCCACACCAGAATCAGGG Hsap_GATA1_ex2_gR40f       BslI CCNNNNNNNGG
## 5  CCAGAGCAGGATCCACAAACTGG  Hsap_GATA1_ex2_gR7r       BslI CCNNNNNNNGG
## 6  GTGTCCTCCACACCAGAATCAGG Hsap_GATA1_ex2_gR39f      HinfI       GANTC
## 7  TGTCCTCCACACCAGAATCAGGG Hsap_GATA1_ex2_gR40f      HinfI       GANTC
## 8  CCAGAGCAGGATCCACAAACTGG  Hsap_GATA1_ex2_gR7r      PflMI CCANNNNNTGG
## 9  GTGTCCTCCACACCAGAATCAGG Hsap_GATA1_ex2_gR39f       TfiI       GAWTC
## 10 TGTCCTCCACACCAGAATCAGGG Hsap_GATA1_ex2_gR40f       TfiI       GAWTC
##    REcutStart REcutEnd
## 2          14       19
## 3          13       23
## 4          12       22
## 5          13       23
## 6          16       20
## 7          15       19
## 8          13       23
## 9          16       20
## 10         15       19
## 
## $REs.isUnique100
## [1] ""           ""           "HinfI TfiI" "HinfI TfiI" "PflMI"     
## 
## $REs.isUnique50
## [1] ""           ""           "HinfI TfiI" "HinfI TfiI" "PflMI"</code></pre>
<p>Paired nickases decreases off-target cleavage by requiring the independent binding of two separate gRNAs around a genomic region. Here is how to find gRNAs in paired configuration.</p>
</div>
<div id="scenario-2-find-grnas-in-paired-configuration-and-perform-off-target-analysis-for-each-grna" class="section level1">
<h1 class="hasAnchor">
<a href="#scenario-2-find-grnas-in-paired-configuration-and-perform-off-target-analysis-for-each-grna" class="anchor"></a>Scenario 2: Find gRNAs in paired configuration and perform off-target analysis for each gRNA</h1>
<div class="sourceCode" id="cb36"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/CRISPRseek/man/offTargetAnalysis.html">offTargetAnalysis</a></span>(<span class="no">inputFilePath</span>,
    <span class="kw">findPairedgRNAOnly</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
    <span class="kw">min.gap</span> <span class="kw">=</span> <span class="fl">0</span>,
    <span class="kw">max.gap</span> <span class="kw">=</span> <span class="fl">20</span>,
    <span class="kw">BSgenomeName</span> <span class="kw">=</span> <span class="no">Hsapiens</span>,
    <span class="kw">chromToSearch</span> <span class="kw">=</span><span class="st">"chrX"</span>,
    <span class="kw">txdb</span> <span class="kw">=</span> <span class="no">TxDb.Hsapiens.UCSC.hg19.knownGene</span>,
    <span class="kw">orgAnn</span> <span class="kw">=</span> <span class="no">org.Hs.egSYMBOL</span>,
    <span class="kw">max.mismatch</span> <span class="kw">=</span> <span class="fl">0</span>,
    <span class="kw">outputDir</span> <span class="kw">=</span> <span class="no">outputDir</span>,
    <span class="kw">overwrite</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<pre><code>## Validating input ...
## Searching for gRNAs ...
## &gt;&gt;&gt; Finding all hits in sequence chrX ...
## &gt;&gt;&gt; DONE searching
## Building feature vectors for scoring ...
## Calculating scores ...
## Annotating, filtering and generating reports ...</code></pre>
<pre><code>##   403 genes were dropped because they have exons located on both strands
##   of the same reference sequence or on more than one reference sequence,
##   so cannot be represented by a single genomic range.
##   Use 'single.strand.genes.only=FALSE' to get all the genes in a
##   GRangesList object, or use suppressMessages() to suppress this message.</code></pre>
<pre><code>## Done annotating
## Add paired information...
## Add RE information...
## write gRNAs to bed file...
## Scan for REsites in flanking region...
## Done. Please check output files in directory  /__w/CRISPRseekGUIDEseqBioc2020Workshop/CRISPRseekGUIDEseqBioc2020Workshop/vignettes/CRISPRseekDemo/</code></pre>
<pre><code>## $on.target
##      names                  forViewInUCSC           
## [1,] "Hsap_GATA1_ex2_gR39f" "chrX:48649586-48649608"
## [2,] "Hsap_GATA1_ex2_gR40f" "chrX:48649587-48649609"
## [3,] "Hsap_GATA1_ex2_gR7r"  "chrX:48649564-48649586"
##      extendedSequence                 gRNAefficacy        
## [1,] "TCTGGTGTCCTCCACACCAGAATCAGGGGT" "0.0696295234374132"
## [2,] "CTGGTGTCCTCCACACCAGAATCAGGGGTT" "0.261881755628254" 
## [3,] "GACACCAGAGCAGGATCCACAAACTGGGGG" "0.101719829329578" 
## 
## $summary
##                  names          forViewInUCSC               extendedSequence
## 1 Hsap_GATA1_ex2_gR39f chrX:48649586-48649608 TCTGGTGTCCTCCACACCAGAATCAGGGGT
## 2 Hsap_GATA1_ex2_gR40f chrX:48649587-48649609 CTGGTGTCCTCCACACCAGAATCAGGGGTT
## 3  Hsap_GATA1_ex2_gR7r chrX:48649564-48649586 GACACCAGAGCAGGATCCACAAACTGGGGG
##         gRNAefficacy            gRNAsPlusPAM top5OfftargetTotalScore
## 1 0.0696295234374132 GTGTCCTCCACACCAGAATCNGG                      NA
## 2  0.261881755628254 TGTCCTCCACACCAGAATCANGG                      NA
## 3  0.101719829329578 CCAGAGCAGGATCCACAAACNGG                      NA
##   top10OfftargetTotalScore top1Hit.onTarget.MMdistance2PAM
## 1                       NA                             NMM
## 2                       NA                             NMM
## 3                       NA                             NMM
##   topOfftarget1MMdistance2PAM topOfftarget2MMdistance2PAM
## 1                                                        
## 2                                                        
## 3                                                        
##   topOfftarget3MMdistance2PAM topOfftarget4MMdistance2PAM
## 1                                                        
## 2                                                        
## 3                                                        
##   topOfftarget5MMdistance2PAM topOfftarget6MMdistance2PAM
## 1                                                        
## 2                                                        
## 3                                                        
##   topOfftarget7MMdistance2PAM topOfftarget8MMdistance2PAM
## 1                                                        
## 2                                                        
## 3                                                        
##   topOfftarget9MMdistance2PAM topOfftarget10MMdistance2PAM
## 1                                                         
## 2                                                         
## 3                                                         
##                               PairedgRNAName            REname uniqREin200
## 1                        Hsap_GATA1_ex2_gR7r BslI  HinfI  TfiI            
## 2                        Hsap_GATA1_ex2_gR7r BslI  HinfI  TfiI            
## 3 Hsap_GATA1_ex2_gR39f  Hsap_GATA1_ex2_gR40f       BslI  PflMI            
##   uniqREin100
## 1            
## 2            
## 3            
## 
## $offtarget
##                   name             gRNAPlusPAM       OffTargetSequence inExon
## 1 Hsap_GATA1_ex2_gR39f GTGTCCTCCACACCAGAATCNGG GTGTCCTCCACACCAGAATCAGG   TRUE
## 2 Hsap_GATA1_ex2_gR40f TGTCCTCCACACCAGAATCANGG TGTCCTCCACACCAGAATCAGGG   TRUE
## 3  Hsap_GATA1_ex2_gR7r CCAGAGCAGGATCCACAAACNGG CCAGAGCAGGATCCACAAACTGG   TRUE
##   inIntron entrez_id  gene score n.mismatch mismatch.distance2PAM
## 1               2623 GATA1   100          0                  &lt;NA&gt;
## 2               2623 GATA1   100          0                  &lt;NA&gt;
## 3               2623 GATA1   100          0                  &lt;NA&gt;
##              alignment isCanonicalPAM          forViewInUCSC strand chrom
## 1 ....................              1 chrX:48649586-48649608      +  chrX
## 2 ....................              1 chrX:48649587-48649609      +  chrX
## 3 ....................              1 chrX:48649564-48649586      -  chrX
##   chromStart chromEnd               extendedSequence gRNAefficacy
## 1   48649586 48649608 TCTGGTGTCCTCCACACCAGAATCAGGGGT   0.06962952
## 2   48649587 48649609 CTGGTGTCCTCCACACCAGAATCAGGGGTT   0.26188176
## 3   48649564 48649586 GACACCAGAGCAGGATCCACAAACTGGGGG   0.10171983
##                                                                                                                                                                                                                                                                                                                                                                                                                             flankSequence
## 1 TATGGAGACTGAGGTGATGGAGTGGGAGGAGGGGGAAAGGAGGAAGAGGAGCAGGTGAAAGGATGTGGGGTGGAAGGATTTCTGTGTCTGAGGACCCCTTCTGTCCTCGCAGGTTAATCCCCAGAGGCTCCATGGAGTTCCCTGGCCTGGGGTCCCTGGGGACCTCAGAGCCCCTCCCCCAGTTTGTGGATCCTGCTCTGGTGTCCTCCACACCAGAATCAGGGGTTTTCTTCCCCTCTGGGCCTGAGGGCTTGGATGCAGCAGCTTCCTCCACTGCCCCGAGCACAGCCACCGCTGCAGCTGCGGCACTGGCCTACTACAGGGACGCTGAGGCCTACAGACACTCCCCAGGTAACTCCATTGAGTGGCTGTCTTGGCATTGGCTGAGTGCTGTTGGGGTTGCCATGGAGATCCTTGGCTAGG
## 2 ATGGAGACTGAGGTGATGGAGTGGGAGGAGGGGGAAAGGAGGAAGAGGAGCAGGTGAAAGGATGTGGGGTGGAAGGATTTCTGTGTCTGAGGACCCCTTCTGTCCTCGCAGGTTAATCCCCAGAGGCTCCATGGAGTTCCCTGGCCTGGGGTCCCTGGGGACCTCAGAGCCCCTCCCCCAGTTTGTGGATCCTGCTCTGGTGTCCTCCACACCAGAATCAGGGGTTTTCTTCCCCTCTGGGCCTGAGGGCTTGGATGCAGCAGCTTCCTCCACTGCCCCGAGCACAGCCACCGCTGCAGCTGCGGCACTGGCCTACTACAGGGACGCTGAGGCCTACAGACACTCCCCAGGTAACTCCATTGAGTGGCTGTCTTGGCATTGGCTGAGTGCTGTTGGGGTTGCCATGGAGATCCTTGGCTAGGT
## 3 AACCCCAACAGCACTCAGCCAATGCCAAGACAGCCACTCAATGGAGTTACCTGGGGAGTGTCTGTAGGCCTCAGCGTCCCTGTAGTAGGCCAGTGCCGCAGCTGCAGCGGTGGCTGTGCTCGGGGCAGTGGAGGAAGCTGCTGCATCCAAGCCCTCAGGCCCAGAGGGGAAGAAAACCCCTGATTCTGGTGTGGAGGACACCAGAGCAGGATCCACAAACTGGGGGAGGGGCTCTGAGGTCCCCAGGGACCCCAGGCCAGGGAACTCCATGGAGCCTCTGGGGATTAACCTGCGAGGACAGAAGGGGTCCTCAGACACAGAAATCCTTCCACCCCACATCCTTTCACCTGCTCCTCTTCCTCCTTTCCCCCTCCTCCCACTCCATCACCTCAGTCTCCATATTTCTCCTTCCCACCTCCCCAT
## 
## $gRNAs.bedFormat
##      [,1]   [,2]       [,3]       [,4]                   [,5]              
## [1,] "chrX" "48649585" "48649608" "Hsap_GATA1_ex2_gR39f" "69.6295234374132"
## [2,] "chrX" "48649586" "48649609" "Hsap_GATA1_ex2_gR40f" "261.881755628254"
## [3,] "chrX" "48649563" "48649586" "Hsap_GATA1_ex2_gR7r"  "101.719829329578"
##      [,6] [,7]       [,8]      
## [1,] "+"  "48649601" "48649603"
## [2,] "+"  "48649602" "48649604"
## [3,] "-"  "48649568" "48649570"
## 
## $REcutDetails
##         ReversegRNAPlusPAM     ReversegRNAName      ForwardgRNAPlusPAM
## 1  CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r GTGTCCTCCACACCAGAATCAGG
## 2  CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r GTGTCCTCCACACCAGAATCAGG
## 3  CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r GTGTCCTCCACACCAGAATCAGG
## 4  CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r GTGTCCTCCACACCAGAATCAGG
## 5  CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r GTGTCCTCCACACCAGAATCAGG
## 6  CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r GTGTCCTCCACACCAGAATCAGG
## 7  CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r TGTCCTCCACACCAGAATCAGGG
## 8  CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r TGTCCTCCACACCAGAATCAGGG
## 9  CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r TGTCCTCCACACCAGAATCAGGG
## 10 CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r TGTCCTCCACACCAGAATCAGGG
## 11 CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r TGTCCTCCACACCAGAATCAGGG
## 12 CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r TGTCCTCCACACCAGAATCAGGG
##         ForwardgRNAName gap ForwardREcutgRNAName ForwardREname ForwardREpattern
## 1  Hsap_GATA1_ex2_gR39f   0 Hsap_GATA1_ex2_gR39f          BslI      CCNNNNNNNGG
## 2  Hsap_GATA1_ex2_gR39f   0 Hsap_GATA1_ex2_gR39f          BslI      CCNNNNNNNGG
## 3  Hsap_GATA1_ex2_gR39f   0 Hsap_GATA1_ex2_gR39f         HinfI            GANTC
## 4  Hsap_GATA1_ex2_gR39f   0 Hsap_GATA1_ex2_gR39f         HinfI            GANTC
## 5  Hsap_GATA1_ex2_gR39f   0 Hsap_GATA1_ex2_gR39f          TfiI            GAWTC
## 6  Hsap_GATA1_ex2_gR39f   0 Hsap_GATA1_ex2_gR39f          TfiI            GAWTC
## 7  Hsap_GATA1_ex2_gR40f   1 Hsap_GATA1_ex2_gR40f          BslI      CCNNNNNNNGG
## 8  Hsap_GATA1_ex2_gR40f   1 Hsap_GATA1_ex2_gR40f          BslI      CCNNNNNNNGG
## 9  Hsap_GATA1_ex2_gR40f   1 Hsap_GATA1_ex2_gR40f         HinfI            GANTC
## 10 Hsap_GATA1_ex2_gR40f   1 Hsap_GATA1_ex2_gR40f         HinfI            GANTC
## 11 Hsap_GATA1_ex2_gR40f   1 Hsap_GATA1_ex2_gR40f          TfiI            GAWTC
## 12 Hsap_GATA1_ex2_gR40f   1 Hsap_GATA1_ex2_gR40f          TfiI            GAWTC
##    ForwardREcutStart ForwardREcutEnd ReverseREcutgRNAName ReverseREname
## 1                 13              23  Hsap_GATA1_ex2_gR7r          BslI
## 2                 13              23  Hsap_GATA1_ex2_gR7r         PflMI
## 3                 16              20  Hsap_GATA1_ex2_gR7r          BslI
## 4                 16              20  Hsap_GATA1_ex2_gR7r         PflMI
## 5                 16              20  Hsap_GATA1_ex2_gR7r          BslI
## 6                 16              20  Hsap_GATA1_ex2_gR7r         PflMI
## 7                 12              22  Hsap_GATA1_ex2_gR7r          BslI
## 8                 12              22  Hsap_GATA1_ex2_gR7r         PflMI
## 9                 15              19  Hsap_GATA1_ex2_gR7r          BslI
## 10                15              19  Hsap_GATA1_ex2_gR7r         PflMI
## 11                15              19  Hsap_GATA1_ex2_gR7r          BslI
## 12                15              19  Hsap_GATA1_ex2_gR7r         PflMI
##    ReverseREpattern ReverseREcutStart ReverseREcutEnd
## 1       CCNNNNNNNGG                13              23
## 2       CCANNNNNTGG                13              23
## 3       CCNNNNNNNGG                13              23
## 4       CCANNNNNTGG                13              23
## 5       CCNNNNNNNGG                13              23
## 6       CCANNNNNTGG                13              23
## 7       CCNNNNNNNGG                13              23
## 8       CCANNNNNTGG                13              23
## 9       CCNNNNNNNGG                13              23
## 10      CCANNNNNTGG                13              23
## 11      CCNNNNNNNGG                13              23
## 12      CCANNNNNTGG                13              23
## 
## $REs.isUnique100
## [1] "" "" ""
## 
## $REs.isUnique50
## [1] "" "" ""</code></pre>
<p>You can specify the criteria of off-target search by specifying max.mismatch, e.g., allowing up to 2 mismatches to be considered as potential off-targets, by default it is set to 4.</p>
<div class="sourceCode" id="cb41"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/CRISPRseek/man/offTargetAnalysis.html">offTargetAnalysis</a></span>(<span class="no">inputFilePath</span>,
    <span class="kw">findPairedgRNAOnly</span> <span class="kw">=</span> <span class="fl">TRUE</span>,  <span class="kw">min.gap</span> <span class="kw">=</span> <span class="fl">0</span>, <span class="kw">max.gap</span> <span class="kw">=</span> <span class="fl">20</span>,
    <span class="kw">BSgenomeName</span> <span class="kw">=</span> <span class="no">Hsapiens</span>,
    <span class="kw">chromToSearch</span> <span class="kw">=</span><span class="st">"chrX"</span>,
    <span class="kw">txdb</span> <span class="kw">=</span> <span class="no">TxDb.Hsapiens.UCSC.hg19.knownGene</span>,
    <span class="kw">orgAnn</span> <span class="kw">=</span> <span class="no">org.Hs.egSYMBOL</span>,
    <span class="kw">max.mismatch</span> <span class="kw">=</span> <span class="fl">2</span>,
    <span class="kw">outputDir</span> <span class="kw">=</span> <span class="no">outputDir</span>,
    <span class="kw">overwrite</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<pre><code>## Validating input ...
## Searching for gRNAs ...
## &gt;&gt;&gt; Finding all hits in sequence chrX ...
## &gt;&gt;&gt; DONE searching
## Building feature vectors for scoring ...
## Calculating scores ...
## Annotating, filtering and generating reports ...</code></pre>
<pre><code>##   403 genes were dropped because they have exons located on both strands
##   of the same reference sequence or on more than one reference sequence,
##   so cannot be represented by a single genomic range.
##   Use 'single.strand.genes.only=FALSE' to get all the genes in a
##   GRangesList object, or use suppressMessages() to suppress this message.</code></pre>
<pre><code>## Done annotating
## Add paired information...
## Add RE information...
## write gRNAs to bed file...
## Scan for REsites in flanking region...
## Done. Please check output files in directory  /__w/CRISPRseekGUIDEseqBioc2020Workshop/CRISPRseekGUIDEseqBioc2020Workshop/vignettes/CRISPRseekDemo/</code></pre>
<pre><code>## $on.target
##      names                  forViewInUCSC           
## [1,] "Hsap_GATA1_ex2_gR39f" "chrX:48649586-48649608"
## [2,] "Hsap_GATA1_ex2_gR40f" "chrX:48649587-48649609"
## [3,] "Hsap_GATA1_ex2_gR7r"  "chrX:48649564-48649586"
##      extendedSequence                 gRNAefficacy        
## [1,] "TCTGGTGTCCTCCACACCAGAATCAGGGGT" "0.0696295234374132"
## [2,] "CTGGTGTCCTCCACACCAGAATCAGGGGTT" "0.261881755628254" 
## [3,] "GACACCAGAGCAGGATCCACAAACTGGGGG" "0.101719829329578" 
## 
## $summary
##                  names          forViewInUCSC               extendedSequence
## 1 Hsap_GATA1_ex2_gR39f chrX:48649586-48649608 TCTGGTGTCCTCCACACCAGAATCAGGGGT
## 2 Hsap_GATA1_ex2_gR40f chrX:48649587-48649609 CTGGTGTCCTCCACACCAGAATCAGGGGTT
## 3  Hsap_GATA1_ex2_gR7r chrX:48649564-48649586 GACACCAGAGCAGGATCCACAAACTGGGGG
##         gRNAefficacy            gRNAsPlusPAM top5OfftargetTotalScore
## 1 0.0696295234374132 GTGTCCTCCACACCAGAATCNGG                      NA
## 2  0.261881755628254 TGTCCTCCACACCAGAATCANGG                     2.9
## 3  0.101719829329578 CCAGAGCAGGATCCACAAACNGG                      NA
##   top10OfftargetTotalScore top1Hit.onTarget.MMdistance2PAM
## 1                       NA                             NMM
## 2                      2.9                             NMM
## 3                       NA                             NMM
##   topOfftarget1MMdistance2PAM topOfftarget2MMdistance2PAM
## 1                                                        
## 2                        13,2                            
## 3                                                        
##   topOfftarget3MMdistance2PAM topOfftarget4MMdistance2PAM
## 1                                                        
## 2                                                        
## 3                                                        
##   topOfftarget5MMdistance2PAM topOfftarget6MMdistance2PAM
## 1                                                        
## 2                                                        
## 3                                                        
##   topOfftarget7MMdistance2PAM topOfftarget8MMdistance2PAM
## 1                                                        
## 2                                                        
## 3                                                        
##   topOfftarget9MMdistance2PAM topOfftarget10MMdistance2PAM
## 1                                                         
## 2                                                         
## 3                                                         
##                               PairedgRNAName            REname uniqREin200
## 1                        Hsap_GATA1_ex2_gR7r BslI  HinfI  TfiI            
## 2                        Hsap_GATA1_ex2_gR7r BslI  HinfI  TfiI            
## 3 Hsap_GATA1_ex2_gR39f  Hsap_GATA1_ex2_gR40f       BslI  PflMI            
##   uniqREin100
## 1            
## 2            
## 3            
## 
## $offtarget
##                   name             gRNAPlusPAM       OffTargetSequence inExon
## 1 Hsap_GATA1_ex2_gR39f GTGTCCTCCACACCAGAATCNGG GTGTCCTCCACACCAGAATCAGG   TRUE
## 2 Hsap_GATA1_ex2_gR40f TGTCCTCCACACCAGAATCANGG TGTCCTCAACACCAGAATGATAG       
## 3 Hsap_GATA1_ex2_gR40f TGTCCTCCACACCAGAATCANGG TGTCCTCCACACCAGAATCAGGG   TRUE
## 4  Hsap_GATA1_ex2_gR7r CCAGAGCAGGATCCACAAACNGG CCAGAGCAGGATCCACAAACTGG   TRUE
##   inIntron entrez_id  gene score n.mismatch mismatch.distance2PAM
## 1               2623 GATA1   100          0                      
## 2     TRUE    139324   HDX   2.9          2                  13,2
## 3               2623 GATA1   100          0                      
## 4               2623 GATA1   100          0                      
##              alignment isCanonicalPAM          forViewInUCSC strand chrom
## 1 ....................              1 chrX:48649586-48649608      +  chrX
## 2 .......A..........G.              0 chrX:83738197-83738219      -  chrX
## 3 ....................              1 chrX:48649587-48649609      +  chrX
## 4 ....................              1 chrX:48649564-48649586      -  chrX
##   chromStart chromEnd               extendedSequence gRNAefficacy
## 1   48649586 48649608 TCTGGTGTCCTCCACACCAGAATCAGGGGT   0.06962952
## 2   83738197 83738219                           &lt;NA&gt;           NA
## 3   48649587 48649609 CTGGTGTCCTCCACACCAGAATCAGGGGTT   0.26188176
## 4   48649564 48649586 GACACCAGAGCAGGATCCACAAACTGGGGG   0.10171983
##                                                                                                                                                                                                                                                                                                                                                                                                                             flankSequence
## 1 TATGGAGACTGAGGTGATGGAGTGGGAGGAGGGGGAAAGGAGGAAGAGGAGCAGGTGAAAGGATGTGGGGTGGAAGGATTTCTGTGTCTGAGGACCCCTTCTGTCCTCGCAGGTTAATCCCCAGAGGCTCCATGGAGTTCCCTGGCCTGGGGTCCCTGGGGACCTCAGAGCCCCTCCCCCAGTTTGTGGATCCTGCTCTGGTGTCCTCCACACCAGAATCAGGGGTTTTCTTCCCCTCTGGGCCTGAGGGCTTGGATGCAGCAGCTTCCTCCACTGCCCCGAGCACAGCCACCGCTGCAGCTGCGGCACTGGCCTACTACAGGGACGCTGAGGCCTACAGACACTCCCCAGGTAACTCCATTGAGTGGCTGTCTTGGCATTGGCTGAGTGCTGTTGGGGTTGCCATGGAGATCCTTGGCTAGG
## 2 GCAGGTGCACATAAGTCAAGAATTGAGGTTTGGGAACCTCCACCTAGATTTCAGAGGATGTATGAAAATGTCTGGATGTCCCTGCAGACATTTGCTGCATGGGTGGAGCCCTCATGGAGAACCTCTGCTAGGGCAATGTGGAAGAGACATGTGGGGTTGGAGCCCCAACACAGATAGTGGAGCTGTGAGAAGAGGGCCACTGTCCTCAACACCAGAATGATAGATCCACCGACAGCTTGCACCATGAGCCTGGGAAAGCCACATACACTTAACATCAGTCTGTGAAATCAGCTGAGGGGGGCGCTGTACCCTGCAAAGCCACAAAGTTGGAGCTGCCCAAGCCCTTTGATGCCCACTCCTTGCATCAGCATAACCTGAATGTGGGACATGGAGTCAAAGGAGATCACTTTGGAACTTTAAGTT
## 3 ATGGAGACTGAGGTGATGGAGTGGGAGGAGGGGGAAAGGAGGAAGAGGAGCAGGTGAAAGGATGTGGGGTGGAAGGATTTCTGTGTCTGAGGACCCCTTCTGTCCTCGCAGGTTAATCCCCAGAGGCTCCATGGAGTTCCCTGGCCTGGGGTCCCTGGGGACCTCAGAGCCCCTCCCCCAGTTTGTGGATCCTGCTCTGGTGTCCTCCACACCAGAATCAGGGGTTTTCTTCCCCTCTGGGCCTGAGGGCTTGGATGCAGCAGCTTCCTCCACTGCCCCGAGCACAGCCACCGCTGCAGCTGCGGCACTGGCCTACTACAGGGACGCTGAGGCCTACAGACACTCCCCAGGTAACTCCATTGAGTGGCTGTCTTGGCATTGGCTGAGTGCTGTTGGGGTTGCCATGGAGATCCTTGGCTAGGT
## 4 AACCCCAACAGCACTCAGCCAATGCCAAGACAGCCACTCAATGGAGTTACCTGGGGAGTGTCTGTAGGCCTCAGCGTCCCTGTAGTAGGCCAGTGCCGCAGCTGCAGCGGTGGCTGTGCTCGGGGCAGTGGAGGAAGCTGCTGCATCCAAGCCCTCAGGCCCAGAGGGGAAGAAAACCCCTGATTCTGGTGTGGAGGACACCAGAGCAGGATCCACAAACTGGGGGAGGGGCTCTGAGGTCCCCAGGGACCCCAGGCCAGGGAACTCCATGGAGCCTCTGGGGATTAACCTGCGAGGACAGAAGGGGTCCTCAGACACAGAAATCCTTCCACCCCACATCCTTTCACCTGCTCCTCTTCCTCCTTTCCCCCTCCTCCCACTCCATCACCTCAGTCTCCATATTTCTCCTTCCCACCTCCCCAT
## 
## $gRNAs.bedFormat
##      [,1]   [,2]       [,3]       [,4]                   [,5]              
## [1,] "chrX" "48649585" "48649608" "Hsap_GATA1_ex2_gR39f" "69.6295234374132"
## [2,] "chrX" "48649586" "48649609" "Hsap_GATA1_ex2_gR40f" "261.881755628254"
## [3,] "chrX" "48649563" "48649586" "Hsap_GATA1_ex2_gR7r"  "101.719829329578"
##      [,6] [,7]       [,8]      
## [1,] "+"  "48649601" "48649603"
## [2,] "+"  "48649602" "48649604"
## [3,] "-"  "48649568" "48649570"
## 
## $REcutDetails
##         ReversegRNAPlusPAM     ReversegRNAName      ForwardgRNAPlusPAM
## 1  CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r GTGTCCTCCACACCAGAATCAGG
## 2  CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r GTGTCCTCCACACCAGAATCAGG
## 3  CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r GTGTCCTCCACACCAGAATCAGG
## 4  CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r GTGTCCTCCACACCAGAATCAGG
## 5  CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r GTGTCCTCCACACCAGAATCAGG
## 6  CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r GTGTCCTCCACACCAGAATCAGG
## 7  CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r TGTCCTCCACACCAGAATCAGGG
## 8  CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r TGTCCTCCACACCAGAATCAGGG
## 9  CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r TGTCCTCCACACCAGAATCAGGG
## 10 CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r TGTCCTCCACACCAGAATCAGGG
## 11 CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r TGTCCTCCACACCAGAATCAGGG
## 12 CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r TGTCCTCCACACCAGAATCAGGG
##         ForwardgRNAName gap ForwardREcutgRNAName ForwardREname ForwardREpattern
## 1  Hsap_GATA1_ex2_gR39f   0 Hsap_GATA1_ex2_gR39f          BslI      CCNNNNNNNGG
## 2  Hsap_GATA1_ex2_gR39f   0 Hsap_GATA1_ex2_gR39f          BslI      CCNNNNNNNGG
## 3  Hsap_GATA1_ex2_gR39f   0 Hsap_GATA1_ex2_gR39f         HinfI            GANTC
## 4  Hsap_GATA1_ex2_gR39f   0 Hsap_GATA1_ex2_gR39f         HinfI            GANTC
## 5  Hsap_GATA1_ex2_gR39f   0 Hsap_GATA1_ex2_gR39f          TfiI            GAWTC
## 6  Hsap_GATA1_ex2_gR39f   0 Hsap_GATA1_ex2_gR39f          TfiI            GAWTC
## 7  Hsap_GATA1_ex2_gR40f   1 Hsap_GATA1_ex2_gR40f          BslI      CCNNNNNNNGG
## 8  Hsap_GATA1_ex2_gR40f   1 Hsap_GATA1_ex2_gR40f          BslI      CCNNNNNNNGG
## 9  Hsap_GATA1_ex2_gR40f   1 Hsap_GATA1_ex2_gR40f         HinfI            GANTC
## 10 Hsap_GATA1_ex2_gR40f   1 Hsap_GATA1_ex2_gR40f         HinfI            GANTC
## 11 Hsap_GATA1_ex2_gR40f   1 Hsap_GATA1_ex2_gR40f          TfiI            GAWTC
## 12 Hsap_GATA1_ex2_gR40f   1 Hsap_GATA1_ex2_gR40f          TfiI            GAWTC
##    ForwardREcutStart ForwardREcutEnd ReverseREcutgRNAName ReverseREname
## 1                 13              23  Hsap_GATA1_ex2_gR7r          BslI
## 2                 13              23  Hsap_GATA1_ex2_gR7r         PflMI
## 3                 16              20  Hsap_GATA1_ex2_gR7r          BslI
## 4                 16              20  Hsap_GATA1_ex2_gR7r         PflMI
## 5                 16              20  Hsap_GATA1_ex2_gR7r          BslI
## 6                 16              20  Hsap_GATA1_ex2_gR7r         PflMI
## 7                 12              22  Hsap_GATA1_ex2_gR7r          BslI
## 8                 12              22  Hsap_GATA1_ex2_gR7r         PflMI
## 9                 15              19  Hsap_GATA1_ex2_gR7r          BslI
## 10                15              19  Hsap_GATA1_ex2_gR7r         PflMI
## 11                15              19  Hsap_GATA1_ex2_gR7r          BslI
## 12                15              19  Hsap_GATA1_ex2_gR7r         PflMI
##    ReverseREpattern ReverseREcutStart ReverseREcutEnd
## 1       CCNNNNNNNGG                13              23
## 2       CCANNNNNTGG                13              23
## 3       CCNNNNNNNGG                13              23
## 4       CCANNNNNTGG                13              23
## 5       CCNNNNNNNGG                13              23
## 6       CCANNNNNTGG                13              23
## 7       CCNNNNNNNGG                13              23
## 8       CCANNNNNTGG                13              23
## 9       CCNNNNNNNGG                13              23
## 10      CCANNNNNTGG                13              23
## 11      CCNNNNNNNGG                13              23
## 12      CCANNNNNTGG                13              23
## 
## $REs.isUnique100
## [1] "" "" ""
## 
## $REs.isUnique50
## [1] "" "" ""</code></pre>
<p>Paired gRNAs in proper spacing and orientation give more specificity and gRNAs overlap with restriction enzyme cut sites facilitates cleavage monitoring. Calling the function offTargetAnalysis with findPairedgRNAOnly = TRUE and findgRNAsWithREcutOnly = TRUE results in searching, scoring and annotating gRNAs that are in paired configuration and at least one of the pairs overlap a restriction enzyme cut site. To be considered as a pair, gap between forward gRNA and the corresponding reverse gRNA needs to be (min.gap, max.gap) inclusive and the reverse gRNA must sit before the forward gRNA. The default (min.gap, max.gap) is (0,20). In order for a gRNA to be considered overlap with restriction enzyme cut site, the enzyme cut pattern must overlap with one of the gRNA positions specified in overlap.gRNA.positions, default position 17 and 18.</p>
</div>
<div id="scenario-3-find-paired-grnas-with-restriction-enzyme-cut-sites-and-perform-off-target-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#scenario-3-find-paired-grnas-with-restriction-enzyme-cut-sites-and-perform-off-target-analysis" class="anchor"></a>Scenario 3: Find paired gRNAs with restriction enzyme cut site(s) and perform off-target analysis</h1>
<div class="sourceCode" id="cb46"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/CRISPRseek/man/offTargetAnalysis.html">offTargetAnalysis</a></span>(<span class="no">inputFilePath</span>,
    <span class="kw">findgRNAsWithREcutOnly</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
        <span class="kw">minREpatternSize</span> <span class="kw">=</span> <span class="fl">6</span>,
    <span class="kw">overlap.gRNA.positions</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">17</span>, <span class="fl">18</span>),
    <span class="kw">findPairedgRNAOnly</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">min.gap</span> <span class="kw">=</span> <span class="fl">0</span>, <span class="kw">max.gap</span> <span class="kw">=</span> <span class="fl">20</span>,
        <span class="kw">BSgenomeName</span> <span class="kw">=</span> <span class="no">Hsapiens</span>,
    <span class="kw">chromToSearch</span> <span class="kw">=</span><span class="st">"chrX"</span>,
        <span class="kw">txdb</span> <span class="kw">=</span> <span class="no">TxDb.Hsapiens.UCSC.hg19.knownGene</span>,
    <span class="kw">orgAnn</span> <span class="kw">=</span> <span class="no">org.Hs.egSYMBOL</span>,
    <span class="kw">max.mismatch</span> <span class="kw">=</span> <span class="fl">0</span>,
        <span class="kw">outputDir</span> <span class="kw">=</span> <span class="no">outputDir</span>,
    <span class="kw">overwrite</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<pre><code>## Validating input ...
## Searching for gRNAs ...
## &gt;&gt;&gt; Finding all hits in sequence chrX ...
## &gt;&gt;&gt; DONE searching
## Building feature vectors for scoring ...
## Calculating scores ...
## Annotating, filtering and generating reports ...</code></pre>
<pre><code>##   403 genes were dropped because they have exons located on both strands
##   of the same reference sequence or on more than one reference sequence,
##   so cannot be represented by a single genomic range.
##   Use 'single.strand.genes.only=FALSE' to get all the genes in a
##   GRangesList object, or use suppressMessages() to suppress this message.</code></pre>
<pre><code>## Done annotating
## Add paired information...
## Add RE information...
## write gRNAs to bed file...
## Scan for REsites in flanking region...
## Done. Please check output files in directory  /__w/CRISPRseekGUIDEseqBioc2020Workshop/CRISPRseekGUIDEseqBioc2020Workshop/vignettes/CRISPRseekDemo/</code></pre>
<pre><code>## $on.target
##      names                  forViewInUCSC           
## [1,] "Hsap_GATA1_ex2_gR39f" "chrX:48649586-48649608"
## [2,] "Hsap_GATA1_ex2_gR40f" "chrX:48649587-48649609"
## [3,] "Hsap_GATA1_ex2_gR7r"  "chrX:48649564-48649586"
##      extendedSequence                 gRNAefficacy        
## [1,] "TCTGGTGTCCTCCACACCAGAATCAGGGGT" "0.0696295234374132"
## [2,] "CTGGTGTCCTCCACACCAGAATCAGGGGTT" "0.261881755628254" 
## [3,] "GACACCAGAGCAGGATCCACAAACTGGGGG" "0.101719829329578" 
## 
## $summary
##                  names          forViewInUCSC               extendedSequence
## 1 Hsap_GATA1_ex2_gR39f chrX:48649586-48649608 TCTGGTGTCCTCCACACCAGAATCAGGGGT
## 2 Hsap_GATA1_ex2_gR40f chrX:48649587-48649609 CTGGTGTCCTCCACACCAGAATCAGGGGTT
## 3  Hsap_GATA1_ex2_gR7r chrX:48649564-48649586 GACACCAGAGCAGGATCCACAAACTGGGGG
##         gRNAefficacy            gRNAsPlusPAM top5OfftargetTotalScore
## 1 0.0696295234374132 GTGTCCTCCACACCAGAATCNGG                      NA
## 2  0.261881755628254 TGTCCTCCACACCAGAATCANGG                      NA
## 3  0.101719829329578 CCAGAGCAGGATCCACAAACNGG                      NA
##   top10OfftargetTotalScore top1Hit.onTarget.MMdistance2PAM
## 1                       NA                             NMM
## 2                       NA                             NMM
## 3                       NA                             NMM
##   topOfftarget1MMdistance2PAM topOfftarget2MMdistance2PAM
## 1                                                        
## 2                                                        
## 3                                                        
##   topOfftarget3MMdistance2PAM topOfftarget4MMdistance2PAM
## 1                                                        
## 2                                                        
## 3                                                        
##   topOfftarget5MMdistance2PAM topOfftarget6MMdistance2PAM
## 1                                                        
## 2                                                        
## 3                                                        
##   topOfftarget7MMdistance2PAM topOfftarget8MMdistance2PAM
## 1                                                        
## 2                                                        
## 3                                                        
##   topOfftarget9MMdistance2PAM topOfftarget10MMdistance2PAM
## 1                                                         
## 2                                                         
## 3                                                         
##                               PairedgRNAName      REname uniqREin200
## 1                        Hsap_GATA1_ex2_gR7r        BslI            
## 2                        Hsap_GATA1_ex2_gR7r        BslI            
## 3 Hsap_GATA1_ex2_gR39f  Hsap_GATA1_ex2_gR40f BslI  PflMI            
##   uniqREin100
## 1            
## 2            
## 3            
## 
## $offtarget
##                   name             gRNAPlusPAM       OffTargetSequence inExon
## 1 Hsap_GATA1_ex2_gR39f GTGTCCTCCACACCAGAATCNGG GTGTCCTCCACACCAGAATCAGG   TRUE
## 2 Hsap_GATA1_ex2_gR40f TGTCCTCCACACCAGAATCANGG TGTCCTCCACACCAGAATCAGGG   TRUE
## 3  Hsap_GATA1_ex2_gR7r CCAGAGCAGGATCCACAAACNGG CCAGAGCAGGATCCACAAACTGG   TRUE
##   inIntron entrez_id  gene score n.mismatch mismatch.distance2PAM
## 1               2623 GATA1   100          0                  &lt;NA&gt;
## 2               2623 GATA1   100          0                  &lt;NA&gt;
## 3               2623 GATA1   100          0                  &lt;NA&gt;
##              alignment isCanonicalPAM          forViewInUCSC strand chrom
## 1 ....................              1 chrX:48649586-48649608      +  chrX
## 2 ....................              1 chrX:48649587-48649609      +  chrX
## 3 ....................              1 chrX:48649564-48649586      -  chrX
##   chromStart chromEnd               extendedSequence gRNAefficacy
## 1   48649586 48649608 TCTGGTGTCCTCCACACCAGAATCAGGGGT   0.06962952
## 2   48649587 48649609 CTGGTGTCCTCCACACCAGAATCAGGGGTT   0.26188176
## 3   48649564 48649586 GACACCAGAGCAGGATCCACAAACTGGGGG   0.10171983
##                                                                                                                                                                                                                                                                                                                                                                                                                             flankSequence
## 1 TATGGAGACTGAGGTGATGGAGTGGGAGGAGGGGGAAAGGAGGAAGAGGAGCAGGTGAAAGGATGTGGGGTGGAAGGATTTCTGTGTCTGAGGACCCCTTCTGTCCTCGCAGGTTAATCCCCAGAGGCTCCATGGAGTTCCCTGGCCTGGGGTCCCTGGGGACCTCAGAGCCCCTCCCCCAGTTTGTGGATCCTGCTCTGGTGTCCTCCACACCAGAATCAGGGGTTTTCTTCCCCTCTGGGCCTGAGGGCTTGGATGCAGCAGCTTCCTCCACTGCCCCGAGCACAGCCACCGCTGCAGCTGCGGCACTGGCCTACTACAGGGACGCTGAGGCCTACAGACACTCCCCAGGTAACTCCATTGAGTGGCTGTCTTGGCATTGGCTGAGTGCTGTTGGGGTTGCCATGGAGATCCTTGGCTAGG
## 2 ATGGAGACTGAGGTGATGGAGTGGGAGGAGGGGGAAAGGAGGAAGAGGAGCAGGTGAAAGGATGTGGGGTGGAAGGATTTCTGTGTCTGAGGACCCCTTCTGTCCTCGCAGGTTAATCCCCAGAGGCTCCATGGAGTTCCCTGGCCTGGGGTCCCTGGGGACCTCAGAGCCCCTCCCCCAGTTTGTGGATCCTGCTCTGGTGTCCTCCACACCAGAATCAGGGGTTTTCTTCCCCTCTGGGCCTGAGGGCTTGGATGCAGCAGCTTCCTCCACTGCCCCGAGCACAGCCACCGCTGCAGCTGCGGCACTGGCCTACTACAGGGACGCTGAGGCCTACAGACACTCCCCAGGTAACTCCATTGAGTGGCTGTCTTGGCATTGGCTGAGTGCTGTTGGGGTTGCCATGGAGATCCTTGGCTAGGT
## 3 AACCCCAACAGCACTCAGCCAATGCCAAGACAGCCACTCAATGGAGTTACCTGGGGAGTGTCTGTAGGCCTCAGCGTCCCTGTAGTAGGCCAGTGCCGCAGCTGCAGCGGTGGCTGTGCTCGGGGCAGTGGAGGAAGCTGCTGCATCCAAGCCCTCAGGCCCAGAGGGGAAGAAAACCCCTGATTCTGGTGTGGAGGACACCAGAGCAGGATCCACAAACTGGGGGAGGGGCTCTGAGGTCCCCAGGGACCCCAGGCCAGGGAACTCCATGGAGCCTCTGGGGATTAACCTGCGAGGACAGAAGGGGTCCTCAGACACAGAAATCCTTCCACCCCACATCCTTTCACCTGCTCCTCTTCCTCCTTTCCCCCTCCTCCCACTCCATCACCTCAGTCTCCATATTTCTCCTTCCCACCTCCCCAT
## 
## $gRNAs.bedFormat
##      [,1]   [,2]       [,3]       [,4]                   [,5]              
## [1,] "chrX" "48649585" "48649608" "Hsap_GATA1_ex2_gR39f" "69.6295234374132"
## [2,] "chrX" "48649586" "48649609" "Hsap_GATA1_ex2_gR40f" "261.881755628254"
## [3,] "chrX" "48649563" "48649586" "Hsap_GATA1_ex2_gR7r"  "101.719829329578"
##      [,6] [,7]       [,8]      
## [1,] "+"  "48649601" "48649603"
## [2,] "+"  "48649602" "48649604"
## [3,] "-"  "48649568" "48649570"
## 
## $REcutDetails
##        ReversegRNAPlusPAM     ReversegRNAName      ForwardgRNAPlusPAM
## 1 CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r GTGTCCTCCACACCAGAATCAGG
## 2 CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r GTGTCCTCCACACCAGAATCAGG
## 3 CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r TGTCCTCCACACCAGAATCAGGG
## 4 CCAGAGCAGGATCCACAAACTGG Hsap_GATA1_ex2_gR7r TGTCCTCCACACCAGAATCAGGG
##        ForwardgRNAName gap ForwardREcutgRNAName ForwardREname ForwardREpattern
## 1 Hsap_GATA1_ex2_gR39f   0 Hsap_GATA1_ex2_gR39f          BslI      CCNNNNNNNGG
## 2 Hsap_GATA1_ex2_gR39f   0 Hsap_GATA1_ex2_gR39f          BslI      CCNNNNNNNGG
## 3 Hsap_GATA1_ex2_gR40f   1 Hsap_GATA1_ex2_gR40f          BslI      CCNNNNNNNGG
## 4 Hsap_GATA1_ex2_gR40f   1 Hsap_GATA1_ex2_gR40f          BslI      CCNNNNNNNGG
##   ForwardREcutStart ForwardREcutEnd ReverseREcutgRNAName ReverseREname
## 1                13              23  Hsap_GATA1_ex2_gR7r          BslI
## 2                13              23  Hsap_GATA1_ex2_gR7r         PflMI
## 3                12              22  Hsap_GATA1_ex2_gR7r          BslI
## 4                12              22  Hsap_GATA1_ex2_gR7r         PflMI
##   ReverseREpattern ReverseREcutStart ReverseREcutEnd
## 1      CCNNNNNNNGG                13              23
## 2      CCANNNNNTGG                13              23
## 3      CCNNNNNNNGG                13              23
## 4      CCANNNNNTGG                13              23
## 
## $REs.isUnique100
## [1] "" "" ""
## 
## $REs.isUnique50
## [1] "" "" ""</code></pre>
</div>
<div id="scenario-4-find-grnas-with-restriction-enzyme-cut-sites-and-perform-off-target-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#scenario-4-find-grnas-with-restriction-enzyme-cut-sites-and-perform-off-target-analysis" class="anchor"></a>Scenario 4: Find gRNAs with restriction enzyme cut site(s) and perform off-target analysis</h1>
<div class="sourceCode" id="cb51"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/CRISPRseek/man/offTargetAnalysis.html">offTargetAnalysis</a></span>(<span class="no">inputFilePath</span>,
    <span class="kw">findgRNAsWithREcutOnly</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
    <span class="kw">minREpatternSize</span> <span class="kw">=</span> <span class="fl">6</span>,
    <span class="kw">overlap.gRNA.positions</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">17</span>, <span class="fl">18</span>),
    <span class="kw">findPairedgRNAOnly</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
    <span class="kw">BSgenomeName</span> <span class="kw">=</span> <span class="no">Hsapiens</span>,
    <span class="kw">chromToSearch</span> <span class="kw">=</span><span class="st">"chrX"</span>,
    <span class="kw">max.mismatch</span> <span class="kw">=</span> <span class="fl">0</span>,
    <span class="kw">txdb</span> <span class="kw">=</span> <span class="no">TxDb.Hsapiens.UCSC.hg19.knownGene</span>,
    <span class="kw">orgAnn</span> <span class="kw">=</span> <span class="no">org.Hs.egSYMBOL</span>,
    <span class="kw">outputDir</span> <span class="kw">=</span> <span class="no">outputDir</span>,
    <span class="kw">overwrite</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<pre><code>## Validating input ...
## Searching for gRNAs ...
## &gt;&gt;&gt; Finding all hits in sequence chrX ...
## &gt;&gt;&gt; DONE searching
## Building feature vectors for scoring ...
## Calculating scores ...
## Annotating, filtering and generating reports ...</code></pre>
<pre><code>##   403 genes were dropped because they have exons located on both strands
##   of the same reference sequence or on more than one reference sequence,
##   so cannot be represented by a single genomic range.
##   Use 'single.strand.genes.only=FALSE' to get all the genes in a
##   GRangesList object, or use suppressMessages() to suppress this message.</code></pre>
<pre><code>## Done annotating
## Add paired information...
## Add RE information...
## write gRNAs to bed file...
## Scan for REsites in flanking region...
## Done. Please check output files in directory  /__w/CRISPRseekGUIDEseqBioc2020Workshop/CRISPRseekGUIDEseqBioc2020Workshop/vignettes/CRISPRseekDemo/</code></pre>
<pre><code>## $on.target
##      names                  forViewInUCSC           
## [1,] "Hsap_GATA1_ex2_gR20r" "chrX:48649577-48649599"
## [2,] "Hsap_GATA1_ex2_gR39f" "chrX:48649586-48649608"
## [3,] "Hsap_GATA1_ex2_gR40f" "chrX:48649587-48649609"
## [4,] "Hsap_GATA1_ex2_gR7r"  "chrX:48649564-48649586"
##      extendedSequence                 gRNAefficacy        
## [1,] "TTCTGGTGTGGAGGACACCAGAGCAGGATC" "0.142091080678942" 
## [2,] "TCTGGTGTCCTCCACACCAGAATCAGGGGT" "0.0696295234374132"
## [3,] "CTGGTGTCCTCCACACCAGAATCAGGGGTT" "0.261881755628254" 
## [4,] "GACACCAGAGCAGGATCCACAAACTGGGGG" "0.101719829329578" 
## 
## $summary
##                  names          forViewInUCSC               extendedSequence
## 1 Hsap_GATA1_ex2_gR20r chrX:48649577-48649599 TTCTGGTGTGGAGGACACCAGAGCAGGATC
## 2 Hsap_GATA1_ex2_gR39f chrX:48649586-48649608 TCTGGTGTCCTCCACACCAGAATCAGGGGT
## 3 Hsap_GATA1_ex2_gR40f chrX:48649587-48649609 CTGGTGTCCTCCACACCAGAATCAGGGGTT
## 4  Hsap_GATA1_ex2_gR7r chrX:48649564-48649586 GACACCAGAGCAGGATCCACAAACTGGGGG
##         gRNAefficacy            gRNAsPlusPAM top5OfftargetTotalScore
## 1  0.142091080678942 GGTGTGGAGGACACCAGAGCNGG                      NA
## 2 0.0696295234374132 GTGTCCTCCACACCAGAATCNGG                      NA
## 3  0.261881755628254 TGTCCTCCACACCAGAATCANGG                      NA
## 4  0.101719829329578 CCAGAGCAGGATCCACAAACNGG                      NA
##   top10OfftargetTotalScore top1Hit.onTarget.MMdistance2PAM
## 1                       NA                             NMM
## 2                       NA                             NMM
## 3                       NA                             NMM
## 4                       NA                             NMM
##   topOfftarget1MMdistance2PAM topOfftarget2MMdistance2PAM
## 1                                                        
## 2                                                        
## 3                                                        
## 4                                                        
##   topOfftarget3MMdistance2PAM topOfftarget4MMdistance2PAM
## 1                                                        
## 2                                                        
## 3                                                        
## 4                                                        
##   topOfftarget5MMdistance2PAM topOfftarget6MMdistance2PAM
## 1                                                        
## 2                                                        
## 3                                                        
## 4                                                        
##   topOfftarget7MMdistance2PAM topOfftarget8MMdistance2PAM
## 1                                                        
## 2                                                        
## 3                                                        
## 4                                                        
##   topOfftarget9MMdistance2PAM topOfftarget10MMdistance2PAM     REname
## 1                                                          Aco12261II
## 2                                                                BslI
## 3                                                                BslI
## 4                                                          BslI PflMI
##   uniqREin200 uniqREin100
## 1                        
## 2                        
## 3                        
## 4       PflMI       PflMI
## 
## $offtarget
##                   name             gRNAPlusPAM       OffTargetSequence inExon
## 1 Hsap_GATA1_ex2_gR20r GGTGTGGAGGACACCAGAGCNGG GGTGTGGAGGACACCAGAGCAGG   TRUE
## 2 Hsap_GATA1_ex2_gR39f GTGTCCTCCACACCAGAATCNGG GTGTCCTCCACACCAGAATCAGG   TRUE
## 3 Hsap_GATA1_ex2_gR40f TGTCCTCCACACCAGAATCANGG TGTCCTCCACACCAGAATCAGGG   TRUE
## 4  Hsap_GATA1_ex2_gR7r CCAGAGCAGGATCCACAAACNGG CCAGAGCAGGATCCACAAACTGG   TRUE
##   inIntron entrez_id  gene score n.mismatch mismatch.distance2PAM
## 1               2623 GATA1   100          0                  &lt;NA&gt;
## 2               2623 GATA1   100          0                  &lt;NA&gt;
## 3               2623 GATA1   100          0                  &lt;NA&gt;
## 4               2623 GATA1   100          0                  &lt;NA&gt;
##              alignment isCanonicalPAM          forViewInUCSC strand chrom
## 1 ....................              1 chrX:48649577-48649599      -  chrX
## 2 ....................              1 chrX:48649586-48649608      +  chrX
## 3 ....................              1 chrX:48649587-48649609      +  chrX
## 4 ....................              1 chrX:48649564-48649586      -  chrX
##   chromStart chromEnd               extendedSequence gRNAefficacy
## 1   48649577 48649599 TTCTGGTGTGGAGGACACCAGAGCAGGATC   0.14209108
## 2   48649586 48649608 TCTGGTGTCCTCCACACCAGAATCAGGGGT   0.06962952
## 3   48649587 48649609 CTGGTGTCCTCCACACCAGAATCAGGGGTT   0.26188176
## 4   48649564 48649586 GACACCAGAGCAGGATCCACAAACTGGGGG   0.10171983
##                                                                                                                                                                                                                                                                                                                                                                                                                             flankSequence
## 1 GGATCTCCATGGCAACCCCAACAGCACTCAGCCAATGCCAAGACAGCCACTCAATGGAGTTACCTGGGGAGTGTCTGTAGGCCTCAGCGTCCCTGTAGTAGGCCAGTGCCGCAGCTGCAGCGGTGGCTGTGCTCGGGGCAGTGGAGGAAGCTGCTGCATCCAAGCCCTCAGGCCCAGAGGGGAAGAAAACCCCTGATTCTGGTGTGGAGGACACCAGAGCAGGATCCACAAACTGGGGGAGGGGCTCTGAGGTCCCCAGGGACCCCAGGCCAGGGAACTCCATGGAGCCTCTGGGGATTAACCTGCGAGGACAGAAGGGGTCCTCAGACACAGAAATCCTTCCACCCCACATCCTTTCACCTGCTCCTCTTCCTCCTTTCCCCCTCCTCCCACTCCATCACCTCAGTCTCCATATTTCTCCTT
## 2 TATGGAGACTGAGGTGATGGAGTGGGAGGAGGGGGAAAGGAGGAAGAGGAGCAGGTGAAAGGATGTGGGGTGGAAGGATTTCTGTGTCTGAGGACCCCTTCTGTCCTCGCAGGTTAATCCCCAGAGGCTCCATGGAGTTCCCTGGCCTGGGGTCCCTGGGGACCTCAGAGCCCCTCCCCCAGTTTGTGGATCCTGCTCTGGTGTCCTCCACACCAGAATCAGGGGTTTTCTTCCCCTCTGGGCCTGAGGGCTTGGATGCAGCAGCTTCCTCCACTGCCCCGAGCACAGCCACCGCTGCAGCTGCGGCACTGGCCTACTACAGGGACGCTGAGGCCTACAGACACTCCCCAGGTAACTCCATTGAGTGGCTGTCTTGGCATTGGCTGAGTGCTGTTGGGGTTGCCATGGAGATCCTTGGCTAGG
## 3 ATGGAGACTGAGGTGATGGAGTGGGAGGAGGGGGAAAGGAGGAAGAGGAGCAGGTGAAAGGATGTGGGGTGGAAGGATTTCTGTGTCTGAGGACCCCTTCTGTCCTCGCAGGTTAATCCCCAGAGGCTCCATGGAGTTCCCTGGCCTGGGGTCCCTGGGGACCTCAGAGCCCCTCCCCCAGTTTGTGGATCCTGCTCTGGTGTCCTCCACACCAGAATCAGGGGTTTTCTTCCCCTCTGGGCCTGAGGGCTTGGATGCAGCAGCTTCCTCCACTGCCCCGAGCACAGCCACCGCTGCAGCTGCGGCACTGGCCTACTACAGGGACGCTGAGGCCTACAGACACTCCCCAGGTAACTCCATTGAGTGGCTGTCTTGGCATTGGCTGAGTGCTGTTGGGGTTGCCATGGAGATCCTTGGCTAGGT
## 4 AACCCCAACAGCACTCAGCCAATGCCAAGACAGCCACTCAATGGAGTTACCTGGGGAGTGTCTGTAGGCCTCAGCGTCCCTGTAGTAGGCCAGTGCCGCAGCTGCAGCGGTGGCTGTGCTCGGGGCAGTGGAGGAAGCTGCTGCATCCAAGCCCTCAGGCCCAGAGGGGAAGAAAACCCCTGATTCTGGTGTGGAGGACACCAGAGCAGGATCCACAAACTGGGGGAGGGGCTCTGAGGTCCCCAGGGACCCCAGGCCAGGGAACTCCATGGAGCCTCTGGGGATTAACCTGCGAGGACAGAAGGGGTCCTCAGACACAGAAATCCTTCCACCCCACATCCTTTCACCTGCTCCTCTTCCTCCTTTCCCCCTCCTCCCACTCCATCACCTCAGTCTCCATATTTCTCCTTCCCACCTCCCCAT
## 
## $gRNAs.bedFormat
##      [,1]   [,2]       [,3]       [,4]                   [,5]              
## [1,] "chrX" "48649576" "48649599" "Hsap_GATA1_ex2_gR20r" "142.091080678942"
## [2,] "chrX" "48649585" "48649608" "Hsap_GATA1_ex2_gR39f" "69.6295234374132"
## [3,] "chrX" "48649586" "48649609" "Hsap_GATA1_ex2_gR40f" "261.881755628254"
## [4,] "chrX" "48649563" "48649586" "Hsap_GATA1_ex2_gR7r"  "101.719829329578"
##      [,6] [,7]       [,8]      
## [1,] "-"  "48649581" "48649583"
## [2,] "+"  "48649601" "48649603"
## [3,] "+"  "48649602" "48649604"
## [4,] "-"  "48649568" "48649570"
## 
## $REcutDetails
##               gRNAPlusPAM        REcutgRNAName     REname   REpattern
## 2 GGTGTGGAGGACACCAGAGCAGG Hsap_GATA1_ex2_gR20r Aco12261II      CCRGAG
## 3 GTGTCCTCCACACCAGAATCAGG Hsap_GATA1_ex2_gR39f       BslI CCNNNNNNNGG
## 4 TGTCCTCCACACCAGAATCAGGG Hsap_GATA1_ex2_gR40f       BslI CCNNNNNNNGG
## 5 CCAGAGCAGGATCCACAAACTGG  Hsap_GATA1_ex2_gR7r       BslI CCNNNNNNNGG
## 6 CCAGAGCAGGATCCACAAACTGG  Hsap_GATA1_ex2_gR7r      PflMI CCANNNNNTGG
##   REcutStart REcutEnd
## 2         14       19
## 3         13       23
## 4         12       22
## 5         13       23
## 6         13       23
## 
## $REs.isUnique100
## [1] ""      ""      ""      "PflMI"
## 
## $REs.isUnique50
## [1] ""      ""      ""      "PflMI"</code></pre>
</div>
<div id="scenario-5-target-and-off-target-analysis-for-user-specified-grnas" class="section level1">
<h1 class="hasAnchor">
<a href="#scenario-5-target-and-off-target-analysis-for-user-specified-grnas" class="anchor"></a>Scenario 5: Target and off-target analysis for user specified gRNAs</h1>
<p>Calling the function offTargetAnalysis with findgRNAs = FALSE will skip the gRNA search step and go directly to off-target search, scoring and annotation for the input gRNAs. The input gRNAs will be annotated with restriction enzyme cut sites for users to review later. However, paired information will not be available.</p>
<div class="sourceCode" id="cb56"><html><body><pre class="r"><span class="no">gRNAFilePath</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">'extdata'</span>, <span class="st">'testHsap_GATA1_ex2_gRNA1.fa'</span>,
    <span class="kw">package</span> <span class="kw">=</span> <span class="st">'CRISPRseek'</span>)

<span class="fu"><a href="https://rdrr.io/pkg/CRISPRseek/man/offTargetAnalysis.html">offTargetAnalysis</a></span>(<span class="kw">inputFilePath</span> <span class="kw">=</span> <span class="no">gRNAFilePath</span>,
    <span class="kw">findPairedgRNAOnly</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
    <span class="kw">findgRNAs</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
    <span class="kw">BSgenomeName</span> <span class="kw">=</span> <span class="no">Hsapiens</span>,
    <span class="kw">chromToSearch</span> <span class="kw">=</span> <span class="st">'chrX'</span>,
    <span class="kw">txdb</span> <span class="kw">=</span> <span class="no">TxDb.Hsapiens.UCSC.hg19.knownGene</span>,
    <span class="kw">orgAnn</span> <span class="kw">=</span> <span class="no">org.Hs.egSYMBOL</span>,
    <span class="kw">max.mismatch</span> <span class="kw">=</span> <span class="fl">2</span>,
    <span class="kw">outputDir</span> <span class="kw">=</span> <span class="no">outputDir</span>,
    <span class="kw">overwrite</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<pre><code>## Validating input ...
## &gt;&gt;&gt; Finding all hits in sequence chrX ...
## &gt;&gt;&gt; DONE searching
## Building feature vectors for scoring ...
## Calculating scores ...
## Annotating, filtering and generating reports ...</code></pre>
<pre><code>##   403 genes were dropped because they have exons located on both strands
##   of the same reference sequence or on more than one reference sequence,
##   so cannot be represented by a single genomic range.
##   Use 'single.strand.genes.only=FALSE' to get all the genes in a
##   GRangesList object, or use suppressMessages() to suppress this message.</code></pre>
<pre><code>## Done annotating
## Add RE information...
## write gRNAs to bed file...
## Scan for REsites in flanking region...
## Done. Please check output files in directory  /__w/CRISPRseekGUIDEseqBioc2020Workshop/CRISPRseekGUIDEseqBioc2020Workshop/vignettes/CRISPRseekDemo/</code></pre>
<pre><code>## $on.target
##      names                  forViewInUCSC           
## [1,] "Hsap_GATA1_ex2_gRNA1" "chrX:48649564-48649586"
##      extendedSequence                 gRNAefficacy       
## [1,] "TCCCCCAGTTTGTGGATCCTGCTCTGGTGT" "0.106547942216103"
## 
## $summary
##                  names          forViewInUCSC               extendedSequence
## 1 Hsap_GATA1_ex2_gRNA1 chrX:48649564-48649586 TCCCCCAGTTTGTGGATCCTGCTCTGGTGT
##        gRNAefficacy            gRNAsPlusPAM top5OfftargetTotalScore
## 1 0.106547942216103 CCAGTTTGTGGATCCTGCTCNGG                      NA
##   top10OfftargetTotalScore top1Hit.onTarget.MMdistance2PAM
## 1                       NA                             NMM
##   topOfftarget1MMdistance2PAM topOfftarget2MMdistance2PAM
## 1                                                        
##   topOfftarget3MMdistance2PAM topOfftarget4MMdistance2PAM
## 1                                                        
##   topOfftarget5MMdistance2PAM topOfftarget6MMdistance2PAM
## 1                                                        
##   topOfftarget7MMdistance2PAM topOfftarget8MMdistance2PAM
## 1                                                        
##   topOfftarget9MMdistance2PAM topOfftarget10MMdistance2PAM REname uniqREin200
## 1                                                                            
##   uniqREin100
## 1            
## 
## $offtarget
##                      name             gRNAPlusPAM       OffTargetSequence
## chrX Hsap_GATA1_ex2_gRNA1 CCAGTTTGTGGATCCTGCTCNGG CCAGTTTGTGGATCCTGCTCTGG
##      inExon inIntron entrez_id  gene score n.mismatch mismatch.distance2PAM
## chrX   TRUE               2623 GATA1   100          0                  &lt;NA&gt;
##                 alignment isCanonicalPAM          forViewInUCSC strand chrom
## chrX ....................              1 chrX:48649564-48649586      +  chrX
##      chromStart chromEnd               extendedSequence gRNAefficacy
## chrX   48649564 48649586 TCCCCCAGTTTGTGGATCCTGCTCTGGTGT    0.1065479
##                                                                                                                                                                                                                                                                                                                                                                                                                                flankSequence
## chrX ATGGGGAGGTGGGAAGGAGAAATATGGAGACTGAGGTGATGGAGTGGGAGGAGGGGGAAAGGAGGAAGAGGAGCAGGTGAAAGGATGTGGGGTGGAAGGATTTCTGTGTCTGAGGACCCCTTCTGTCCTCGCAGGTTAATCCCCAGAGGCTCCATGGAGTTCCCTGGCCTGGGGTCCCTGGGGACCTCAGAGCCCCTCCCCCAGTTTGTGGATCCTGCTCTGGTGTCCTCCACACCAGAATCAGGGGTTTTCTTCCCCTCTGGGCCTGAGGGCTTGGATGCAGCAGCTTCCTCCACTGCCCCGAGCACAGCCACCGCTGCAGCTGCGGCACTGGCCTACTACAGGGACGCTGAGGCCTACAGACACTCCCCAGGTAACTCCATTGAGTGGCTGTCTTGGCATTGGCTGAGTGCTGTTGGGGTT
## 
## $gRNAs.bedFormat
##      [,1]   [,2]       [,3]       [,4]                   [,5]              
## [1,] "chrX" "48649563" "48649586" "Hsap_GATA1_ex2_gRNA1" "106.547942216103"
##      [,6] [,7]       [,8]      
## [1,] "+"  "48649579" "48649581"
## 
## $REcutDetails
## [1] gRNAPlusPAM   REcutgRNAName REname        REpattern     REcutStart   
## [6] REcutEnd     
## &lt;0 rows&gt; (or 0-length row.names)
## 
## $REs.isUnique100
## [1] ""
## 
## $REs.isUnique50
## [1] ""</code></pre>
</div>
<div id="scenario-6--quick-grna-finding-without-off-target-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#scenario-6--quick-grna-finding-without-off-target-analysis" class="anchor"></a>Scenario 6. Quick gRNA finding without off-target analysis</h1>
<p>Calling the function offTargetAnalysis with chromToSearch = "" results in quick gRNA search without performing off-target analysis. Parameters findgRNAsWithREcutOnly and findPairedgRNAOnly can be tuned to indicate whether searching for gRNAs overlap restriction enzyme cut sites, and whether searching for gRNAs in paired configuration.</p>
<div class="sourceCode" id="cb61"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/CRISPRseek/man/offTargetAnalysis.html">offTargetAnalysis</a></span>(<span class="no">inputFilePath</span>,
    <span class="kw">chromToSearch</span> <span class="kw">=</span> <span class="st">""</span>,
    <span class="kw">outputDir</span> <span class="kw">=</span> <span class="no">outputDir</span>,
    <span class="kw">overwrite</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<pre><code>## Validating input ...
## Searching for gRNAs ...
## Done. Please check output files in directory  /__w/CRISPRseekGUIDEseqBioc2020Workshop/CRISPRseekGUIDEseqBioc2020Workshop/vignettes/CRISPRseekDemo/</code></pre>
<pre><code>## DNAStringSet object of length 5:
##     width seq                                               names               
## [1]    23 CCAGTTTGTGGATCCTGCTCTGG                           Hsap_GATA1_ex2_gR17f
## [2]    23 GTGTCCTCCACACCAGAATCAGG                           Hsap_GATA1_ex2_gR39f
## [3]    23 TGTCCTCCACACCAGAATCAGGG                           Hsap_GATA1_ex2_gR40f
## [4]    23 GGTGTGGAGGACACCAGAGCAGG                           Hsap_GATA1_ex2_gR20r
## [5]    23 CCAGAGCAGGATCCACAAACTGG                           Hsap_GATA1_ex2_gR7r</code></pre>
</div>
<div id="scenario-7--find-potential-grnas-preferentially-targeting-one-of-two-alleles-without-running-time-consuming-off-target-analysis-on-all-possible-grnas-" class="section level1">
<h1 class="hasAnchor">
<a href="#scenario-7--find-potential-grnas-preferentially-targeting-one-of-two-alleles-without-running-time-consuming-off-target-analysis-on-all-possible-grnas-" class="anchor"></a>Scenario 7. Find potential gRNAs preferentially targeting one of two alleles without running time-consuming off-target analysis on all possible gRNAs.</h1>
<p>Below is an example to search for gRNAs that target at least one of the alleles. Two files are provided containing sequences that differ by a single nucleotide polymorphism (SNP). The results are saved in file scoresFor2InputSequences.xls in outputDir directory.</p>
<p>Hungtinton disease is caused by mutations in the HTT gene. Expansion of CAG repeats in one copy of HTT can result in adult onset neurodegeneration. Because HTT is an essential gene, nucleases cannot be used that inactivate both alleles. Therefore, to identify nuclease target sites that are allele-specific, we will try to search for sites that overlap a single nucleotide polymorphism (SNP), RS362331 is located in a coding exon of HTT. Two sequences that differ only at the polymorphism site will be used as inputs for compare2sequences.</p>
<div class="sourceCode" id="cb64"><html><body><pre class="r"><span class="no">inputFile1Path</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"rs362331C.fa"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"CRISPRseek"</span>)
<span class="no">inputFile2Path</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"rs362331T.fa"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"CRISPRseek"</span>)
<span class="no">seqs</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CRISPRseek/man/compare2Sequences.html">compare2Sequences</a></span>(<span class="no">inputFile1Path</span>, <span class="no">inputFile2Path</span>,
    <span class="kw">outputDir</span> <span class="kw">=</span> <span class="no">outputDir</span>,
    <span class="kw">overwrite</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<pre><code>## search for gRNAs for input file1...
## Validating input ...
## Searching for gRNAs ...
## Done. Please check output files in directory  /__w/CRISPRseekGUIDEseqBioc2020Workshop/CRISPRseekGUIDEseqBioc2020Workshop/vignettes/CRISPRseekDemo/rs362331C.fa-Jul-13-2020/ 
## search for gRNAs for input file2...
## Validating input ...
## Searching for gRNAs ...
## Done. Please check output files in directory  /__w/CRISPRseekGUIDEseqBioc2020Workshop/CRISPRseekGUIDEseqBioc2020Workshop/vignettes/CRISPRseekDemo/rs362331T.fa-Jul-13-2020/ 
## [1] "Scoring ..."
## &gt;&gt;&gt; Finding all hits in sequence rs362331T ...
## &gt;&gt;&gt; DONE searching
## finish off-target search in sequence 2
## &gt;&gt;&gt; Finding all hits in sequence rs362331C ...
## &gt;&gt;&gt; DONE searching
## finish off-target search in sequence 1
## finish feature vector building
## finish score calculation
## [1] "Done!"</code></pre>
</div>
<div id="scenario-8--quick-grna-finding-with-grna-efficacy-prediction" class="section level1">
<h1 class="hasAnchor">
<a href="#scenario-8--quick-grna-finding-with-grna-efficacy-prediction" class="anchor"></a>Scenario 8. Quick gRNA finding with gRNA efficacy prediction</h1>
<p>Calling the function offTargetAnalysis with max.mismatch = 0 results in quick gRNA search with gRNA efficacy prediction without off-target analysis.</p>
<div class="sourceCode" id="cb66"><html><body><pre class="r"><span class="no">inputFilePath</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">'extdata'</span>, <span class="st">'inputseq.fa'</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">'CRISPRseek'</span>)
<span class="no">results</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CRISPRseek/man/offTargetAnalysis.html">offTargetAnalysis</a></span>(<span class="no">inputFilePath</span>,
    <span class="kw">annotateExon</span> <span class="kw">=</span> <span class="fl">FALSE</span>,<span class="kw">chromToSearch</span> <span class="kw">=</span> <span class="st">"chrX"</span>,
    <span class="kw">max.mismatch</span> <span class="kw">=</span> <span class="fl">0</span>, <span class="kw">BSgenomeName</span> <span class="kw">=</span> <span class="no">Hsapiens</span>,
    <span class="kw">outputDir</span> <span class="kw">=</span> <span class="no">outputDir</span>, <span class="kw">overwrite</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<pre><code>## Validating input ...
## Searching for gRNAs ...
## &gt;&gt;&gt; Finding all hits in sequence chrX ...
## &gt;&gt;&gt; DONE searching
## Building feature vectors for scoring ...
## Calculating scores ...
## Annotating, filtering and generating reports ...
## Done annotating
## Add paired information...
## Add RE information...
## write gRNAs to bed file...
## Scan for REsites in flanking region...
## Done. Please check output files in directory  /__w/CRISPRseekGUIDEseqBioc2020Workshop/CRISPRseekGUIDEseqBioc2020Workshop/vignettes/CRISPRseekDemo/</code></pre>
<p>Alternatively, you can set useEfficacyFromInputSeq = TRUE and chromToSearch = "" without specifying BSgenomeName if input sequence is long enough.</p>
<div class="sourceCode" id="cb68"><html><body><pre class="r"><span class="no">inputFilePath</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">'extdata'</span>, <span class="st">'inputseq.fa'</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">'CRISPRseek'</span>)
<span class="fu"><a href="https://rdrr.io/pkg/CRISPRseek/man/offTargetAnalysis.html">offTargetAnalysis</a></span>(<span class="no">inputFilePath</span>,
    <span class="kw">annotateExon</span> <span class="kw">=</span> <span class="fl">FALSE</span>,<span class="kw">chromToSearch</span> <span class="kw">=</span> <span class="st">""</span>,
    <span class="kw">useEfficacyFromInputSeq</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
    <span class="kw">max.mismatch</span> <span class="kw">=</span> <span class="fl">0</span>,
    <span class="kw">outputDir</span> <span class="kw">=</span> <span class="no">outputDir</span>, <span class="kw">overwrite</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<pre><code>## Validating input ...
## Searching for gRNAs ...
## Done. Please check output files in directory  /__w/CRISPRseekGUIDEseqBioc2020Workshop/CRISPRseekGUIDEseqBioc2020Workshop/vignettes/CRISPRseekDemo/</code></pre>
<pre><code>## DNAStringSet object of length 5:
##     width seq                                               names               
## [1]    23 CCAGTTTGTGGATCCTGCTCTGG                           Hsap_GATA1_ex2_gR17f
## [2]    23 GTGTCCTCCACACCAGAATCAGG                           Hsap_GATA1_ex2_gR39f
## [3]    23 TGTCCTCCACACCAGAATCAGGG                           Hsap_GATA1_ex2_gR40f
## [4]    23 GGTGTGGAGGACACCAGAGCAGG                           Hsap_GATA1_ex2_gR20r
## [5]    23 CCAGAGCAGGATCCACAAACTGG                           Hsap_GATA1_ex2_gR7r</code></pre>
</div>
<div id="scenario-9--grna-search-and-offtarget-analysis-of-super-long-input-sequence-longer-than-200kb" class="section level1">
<h1 class="hasAnchor">
<a href="#scenario-9--grna-search-and-offtarget-analysis-of-super-long-input-sequence-longer-than-200kb" class="anchor"></a>Scenario 9. gRNA search and offTarget analysis of super long input sequence (longer than 200kb)</h1>
<p>Calling the function offTargetAnalysis with annotatePaired = FALSE, enable.multicore = TRUE and set n.cores.max will improve the performance. We also suggest split the super long sequence into smaller chunks and perform offTarget analysis for each subsequence separately (Thank Alex Williams for sharing this use case at <a href="https://support.bioconductor.org/p/72994/" class="uri">https://support.bioconductor.org/p/72994/</a>). In addition, please remember to use repeat masked sequence as input.</p>
<div class="sourceCode" id="cb71"><html><body><pre class="r"><span class="no">results</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CRISPRseek/man/offTargetAnalysis.html">offTargetAnalysis</a></span>(<span class="no">inputFilePath</span>, <span class="kw">annotatePaired</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
    <span class="kw">chromToSearch</span> <span class="kw">=</span> <span class="st">"chrX"</span>,
    <span class="kw">enable.multicore</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">n.cores.max</span> <span class="kw">=</span> <span class="fl">6</span>,
    <span class="kw">annotateExon</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
    <span class="kw">max.mismatch</span> <span class="kw">=</span> <span class="fl">0</span>, <span class="kw">BSgenomeName</span> <span class="kw">=</span> <span class="no">Hsapiens</span>,
    <span class="kw">outputDir</span> <span class="kw">=</span> <span class="no">outputDir</span>, <span class="kw">overwrite</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<pre><code>## Validating input ...
## Searching for gRNAs ...
## &gt;&gt;&gt; Finding all hits in sequence chrX ...
## &gt;&gt;&gt; DONE searching
## Building feature vectors for scoring ...
## Calculating scores ...
## Annotating, filtering and generating reports ...
## Done annotating
## Add RE information...
## write gRNAs to bed file...
## Scan for REsites in flanking region...
## Done. Please check output files in directory  /__w/CRISPRseekGUIDEseqBioc2020Workshop/CRISPRseekGUIDEseqBioc2020Workshop/vignettes/CRISPRseekDemo/</code></pre>
</div>
<div id="scenario-10--output-cutting-frequency-determination-cfd-score-for-offtargets" class="section level1">
<h1 class="hasAnchor">
<a href="#scenario-10--output-cutting-frequency-determination-cfd-score-for-offtargets" class="anchor"></a>Scenario 10. Output cutting frequency determination (CFD) score for offtargets</h1>
<p>Calling the function offTargetAnalysis with scoring.method set to CFDscore will output CFD score using the algorithm by Doench et al., 2016, which models the effects of both mismatch position and mismatch type on cutting frequency. By default, scoring.method is set to Hsu-Zhang, which only models the effect of mismatch position.</p>
<div class="sourceCode" id="cb73"><html><body><pre class="r"><span class="no">results</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CRISPRseek/man/offTargetAnalysis.html">offTargetAnalysis</a></span>(<span class="no">inputFilePath</span>, <span class="kw">annotatePaired</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
     <span class="kw">scoring.method</span> <span class="kw">=</span> <span class="st">"CFDscore"</span>,
     <span class="kw">chromToSearch</span> <span class="kw">=</span> <span class="st">"chrX"</span>,
     <span class="kw">annotateExon</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
     <span class="kw">max.mismatch</span> <span class="kw">=</span> <span class="fl">2</span>, <span class="kw">BSgenomeName</span> <span class="kw">=</span> <span class="no">Hsapiens</span>,
     <span class="kw">outputDir</span> <span class="kw">=</span> <span class="no">outputDir</span>, <span class="kw">overwrite</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<pre><code>## Validating input ...
## Searching for gRNAs ...
## &gt;&gt;&gt; Finding all hits in sequence chrX ...
## &gt;&gt;&gt; DONE searching
## Building feature vectors for scoring ...
## Calculating scores ...
## Annotating, filtering and generating reports ...
## Done annotating
## Add RE information...
## write gRNAs to bed file...
## Scan for REsites in flanking region...
## Done. Please check output files in directory  /__w/CRISPRseekGUIDEseqBioc2020Workshop/CRISPRseekGUIDEseqBioc2020Workshop/vignettes/CRISPRseekDemo/</code></pre>
</div>
<div id="scenario-11--grna-search-and-offtarget-anlysis-with-pam-on-the-5-prime-side" class="section level1">
<h1 class="hasAnchor">
<a href="#scenario-11--grna-search-and-offtarget-anlysis-with-pam-on-the-5-prime-side" class="anchor"></a>Scenario 11. gRNA search and offtarget anlysis with PAM on the 5 prime side</h1>
<div class="sourceCode" id="cb75"><html><body><pre class="r">  <span class="no">results</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CRISPRseek/man/offTargetAnalysis.html">offTargetAnalysis</a></span>(<span class="no">inputFilePath</span>,
                <span class="kw">annotatePaired</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
                <span class="kw">BSgenomeName</span> <span class="kw">=</span> <span class="no">Hsapiens</span>, <span class="kw">chromToSearch</span> <span class="kw">=</span> <span class="st">"chrX"</span>,
                <span class="kw">txdb</span> <span class="kw">=</span> <span class="no">TxDb.Hsapiens.UCSC.hg19.knownGene</span>,
                <span class="kw">orgAnn</span> <span class="kw">=</span> <span class="no">org.Hs.egSYMBOL</span>, <span class="kw">max.mismatch</span> <span class="kw">=</span> <span class="fl">4</span>,
                <span class="kw">outputDir</span> <span class="kw">=</span> <span class="no">outputDir</span>, <span class="kw">overwrite</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
                <span class="kw">PAM.location</span> <span class="kw">=</span> <span class="st">"5prime"</span>,
                <span class="kw">PAM</span> <span class="kw">=</span> <span class="st">"TGT"</span>,
                <span class="kw">PAM.pattern</span> <span class="kw">=</span> <span class="st">"^T[A|G]N"</span>, <span class="kw">allowed.mismatch.PAM</span> <span class="kw">=</span> <span class="fl">2</span>,
                <span class="kw">subPAM.position</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>,<span class="fl">2</span>))</pre></body></html></div>
<pre><code>## Validating input ...
## Searching for gRNAs ...
## &gt;&gt;&gt; Finding all hits in sequence chrX ...
## &gt;&gt;&gt; DONE searching
## Building feature vectors for scoring ...
## Calculating scores ...
## Annotating, filtering and generating reports ...</code></pre>
<pre><code>##   403 genes were dropped because they have exons located on both strands
##   of the same reference sequence or on more than one reference sequence,
##   so cannot be represented by a single genomic range.
##   Use 'single.strand.genes.only=FALSE' to get all the genes in a
##   GRangesList object, or use suppressMessages() to suppress this message.</code></pre>
<pre><code>## Done annotating
## Add RE information...
## write gRNAs to bed file...
## Scan for REsites in flanking region...
## Done. Please check output files in directory  /__w/CRISPRseekGUIDEseqBioc2020Workshop/CRISPRseekGUIDEseqBioc2020Workshop/vignettes/CRISPRseekDemo/</code></pre>
</div>
<div id="scenario-12--use-crisprscan-to-calculate-efficacy" class="section level1">
<h1 class="hasAnchor">
<a href="#scenario-12--use-crisprscan-to-calculate-efficacy" class="anchor"></a>Scenario 12. Use CRISPRscan to calculate efficacy</h1>
<div class="sourceCode" id="cb79"><html><body><pre class="r"><span class="no">results</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CRISPRseek/man/offTargetAnalysis.html">offTargetAnalysis</a></span>(<span class="no">inputFilePath</span>, <span class="kw">annotatePaired</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
         <span class="kw">annotateExon</span> <span class="kw">=</span> <span class="fl">FALSE</span>,<span class="kw">findPairedgRNAOnly</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">chromToSearch</span> <span class="kw">=</span> <span class="st">"chrX"</span>,
         <span class="kw">max.mismatch</span> <span class="kw">=</span> <span class="fl">0</span>, <span class="kw">BSgenomeName</span> <span class="kw">=</span> <span class="no">Hsapiens</span>,
         <span class="kw">rule.set</span> <span class="kw">=</span> <span class="st">"CRISPRscan"</span>,
         <span class="kw">baseBeforegRNA</span> <span class="kw">=</span> <span class="fl">6</span>,
         <span class="kw">baseAfterPAM</span> <span class="kw">=</span> <span class="fl">6</span>,
         <span class="kw">featureWeightMatrixFile</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"Morenos-Mateo.csv"</span>,
         <span class="kw">package</span> <span class="kw">=</span> <span class="st">"CRISPRseek"</span>),
         <span class="kw">outputDir</span> <span class="kw">=</span> <span class="no">outputDir</span>, <span class="kw">overwrite</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<pre><code>## Validating input ...
## Searching for gRNAs ...
## &gt;&gt;&gt; Finding all hits in sequence chrX ...
## &gt;&gt;&gt; DONE searching
## Building feature vectors for scoring ...
## Calculating scores ...
## Annotating, filtering and generating reports ...
## Done annotating
## Add RE information...
## write gRNAs to bed file...
## Scan for REsites in flanking region...
## Done. Please check output files in directory  /__w/CRISPRseekGUIDEseqBioc2020Workshop/CRISPRseekGUIDEseqBioc2020Workshop/vignettes/CRISPRseekDemo/</code></pre>
</div>
<div id="scenario-13--design-grna-for-base-editing-systems" class="section level1">
<h1 class="hasAnchor">
<a href="#scenario-13--design-grna-for-base-editing-systems" class="anchor"></a>Scenario 13. Design gRNA for base editing systems</h1>
<div class="sourceCode" id="cb81"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/CRISPRseek/man/offTargetAnalysis.html">offTargetAnalysis</a></span>(<span class="no">inputFilePath</span>, <span class="kw">findgRNAsWithREcutOnly</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
         <span class="kw">REpatternFile</span> <span class="kw">=</span> <span class="no">REpatternFile</span>, <span class="kw">findPairedgRNAOnly</span> <span class="kw">=</span> <span class="fl">FALSE</span>,

         <span class="kw">annotatePaired</span> <span class="kw">=</span> <span class="fl">FALSE</span>,

         <span class="kw">BSgenomeName</span> <span class="kw">=</span> <span class="no">Hsapiens</span>, <span class="kw">chromToSearch</span> <span class="kw">=</span> <span class="st">"chrX"</span>,

         <span class="kw">txdb</span> <span class="kw">=</span> <span class="no">TxDb.Hsapiens.UCSC.hg19.knownGene</span>,

         <span class="kw">orgAnn</span> <span class="kw">=</span> <span class="no">org.Hs.egSYMBOL</span>, <span class="kw">max.mismatch</span> <span class="kw">=</span> <span class="fl">4</span>,

         <span class="kw">outputDir</span> <span class="kw">=</span> <span class="no">outputDir</span>, <span class="kw">overwrite</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">PAM.location</span> <span class="kw">=</span> <span class="st">"5prime"</span>,

         <span class="kw">PAM</span> <span class="kw">=</span> <span class="st">"TGT"</span>, <span class="kw">PAM.pattern</span> <span class="kw">=</span> <span class="st">"^T[A|G]N"</span>, <span class="kw">allowed.mismatch.PAM</span> <span class="kw">=</span> <span class="fl">2</span>,

         <span class="kw">subPAM.position</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>,<span class="fl">2</span>), <span class="kw">baseEditing</span> <span class="kw">=</span> <span class="fl">TRUE</span>,

         <span class="kw">editingWindow</span> <span class="kw">=</span> <span class="fl">10</span>:<span class="fl">20</span>, <span class="kw">targetBase</span> <span class="kw">=</span> <span class="st">"A"</span>)</pre></body></html></div>
<pre><code>## Validating input ...
## Searching for gRNAs ...
## No gRNAs found in the input sequence Hsap_GATA1_ex2no gRNAs found!</code></pre>
</div>
<div id="scenario-14--design-grna-for-prime-editing-systems" class="section level1">
<h1 class="hasAnchor">
<a href="#scenario-14--design-grna-for-prime-editing-systems" class="anchor"></a>Scenario 14. Design gRNA for prime editing systems</h1>
<div class="sourceCode" id="cb83"><html><body><pre class="r"><span class="no">inputFilePath</span> <span class="kw">&lt;-</span> <span class="fu">DNAStringSet</span>(<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(
<span class="st">"CCAGTTTGTGGATCCTGCTCTGGTGTCCTCCACACCAGAATCAGGGATCGAAAA"</span>,
<span class="st">"CTCATCAGTCGATGCGAGTCATCTAAATTCCGATCAATTTCACACTTTAAACG"</span>, <span class="kw">sep</span> <span class="kw">=</span><span class="st">""</span>))
        <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">inputFilePath</span>) <span class="kw">&lt;-</span> <span class="st">"testPE"</span>
        <span class="no">results3</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CRISPRseek/man/offTargetAnalysis.html">offTargetAnalysis</a></span>(<span class="no">inputFilePath</span>,
            <span class="kw">gRNAoutputName</span> <span class="kw">=</span> <span class="st">"testPEgRNAs"</span>,
            <span class="kw">BSgenomeName</span> <span class="kw">=</span> <span class="no">Hsapiens</span>, <span class="kw">chromToSearch</span> <span class="kw">=</span> <span class="st">"chrX"</span>,
            <span class="kw">txdb</span> <span class="kw">=</span> <span class="no">TxDb.Hsapiens.UCSC.hg19.knownGene</span>,
            <span class="kw">orgAnn</span> <span class="kw">=</span> <span class="no">org.Hs.egSYMBOL</span>, <span class="kw">max.mismatch</span> <span class="kw">=</span> <span class="fl">1</span>,
            <span class="kw">outputDir</span> <span class="kw">=</span> <span class="no">outputDir</span>, <span class="kw">overwrite</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
            <span class="kw">PAM.size</span> <span class="kw">=</span> <span class="fl">3L</span>,
            <span class="kw">gRNA.size</span> <span class="kw">=</span> <span class="fl">20L</span>,
            <span class="kw">overlap.gRNA.positions</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">17L</span>,<span class="fl">18L</span>),
            <span class="kw">PBS.length</span> <span class="kw">=</span> <span class="fl">15</span>,
            <span class="kw">corrected.seq</span> <span class="kw">=</span> <span class="st">"T"</span>,
            <span class="kw">RT.template.pattern</span> <span class="kw">=</span> <span class="st">"D$"</span>,
            <span class="kw">RT.template.length</span> <span class="kw">=</span> <span class="fl">8</span>:<span class="fl">30</span>,
            <span class="kw">targeted.seq.length.change</span> <span class="kw">=</span> <span class="fl">0</span>,
            <span class="kw">bp.after.target.end</span> <span class="kw">=</span> <span class="fl">15</span>,
            <span class="kw">target.start</span> <span class="kw">=</span> <span class="fl">20</span>,
            <span class="kw">target.end</span> <span class="kw">=</span> <span class="fl">20</span>,
            <span class="kw">paired.orientation</span> <span class="kw">=</span> <span class="st">"PAMin"</span>, <span class="kw">min.gap</span> <span class="kw">=</span> <span class="fl">20</span>, <span class="kw">max.gap</span> <span class="kw">=</span> <span class="fl">90</span>,
            <span class="kw">primeEditing</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">findPairedgRNAOnly</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<pre><code>## Validating input ...
## Searching for gRNAs ...
## &gt;&gt;&gt; Finding all hits in sequence chrX ...
## &gt;&gt;&gt; DONE searching
## Building feature vectors for scoring ...
## Calculating scores ...
## Annotating, filtering and generating reports ...</code></pre>
<pre><code>##   403 genes were dropped because they have exons located on both strands
##   of the same reference sequence or on more than one reference sequence,
##   so cannot be represented by a single genomic range.
##   Use 'single.strand.genes.only=FALSE' to get all the genes in a
##   GRangesList object, or use suppressMessages() to suppress this message.</code></pre>
<pre><code>## Done annotating
## Add paired information...
## Add RE information...
## write gRNAs to bed file...
## Scan for REsites in flanking region...
## Done. Please check output files in directory  /__w/CRISPRseekGUIDEseqBioc2020Workshop/CRISPRseekGUIDEseqBioc2020Workshop/vignettes/CRISPRseekDemo/</code></pre>
</div>
<div id="exercise-1" class="section level1">
<h1 class="hasAnchor">
<a href="#exercise-1" class="anchor"></a>Exercise 1</h1>
<p>To preferentially target one allele, select gRNAs that have the lowest score for the other allele. Selected gRNAs can then be examined for potential off-target cleavage as described in Scenario 5.</p>
</div>
<div id="exercise-2" class="section level1">
<h1 class="hasAnchor">
<a href="#exercise-2" class="anchor"></a>Exercise 2</h1>
<p>Identify gRNAs that target the following two input sequences equally well with minimized off-target cleavage</p>
<blockquote>
<p>MfSerpAEx2<br>
GACGATGGCATCCTCCGTTCCCTGGGGCCTCCTGCTGCTGGCGGGGCTGTGCTGCCTGGCCCCCCGCTCCCTGGCCTCGAGTCCCCTGGGAGCCGCTGTCCAGGACACAGGTGCACCCCACCACGACCATGAGCACCATGAGGAGCCAGCCTGCCACAAGATTGCCCCGAACCTGGCCGACTTCGCCTTCAGCATGTACCGCCAGGTGGCGCATGGGTCCAACACCACCAACATCTTCTTCTCCCCCGTGAGCATCGCGACCGCCTTTGCGTTGCTTTCTCTGGGGGCCAAGGGTGACACTCACTCCGAGATCATGAAGGGCCTTAGGTTCAACCTCACTGAGAGAGCCGAGGGTGAGGTCCACCAAGGCTTCCAGCAACTTCTCCGCACCCTCAACCACCCAGACAACCAGCTGCAGCTGACCACTGGCAATGGTCTCTTCATCGCTGAGGGCATGAAGCTACTGGATAAGTTTTTGGAGGATGTCAAGAACCTGTACCACTCAGAAGCCTTCTCCACCAATTTCGGGGACACCGAAGCAGCCAAGAAACAGATCAACGATTATGTTGAGAAGGGAACCCAAGGGAAAATTGTGGATTTGGTCAAAGACCTTGACAAAGACACAGCTTTCGCTCTGGTGAATTACATTTTCTTTAAAG</p>
</blockquote>
<blockquote>
<p>HsSerpAEx2<br>
GACAATGCCGTCTTCTGTCTCGTGGGGCATCCTCCTGCTGGCAGGCCTGTGCTGCCTGGTCCCTGTCTCCCTGGCTGAGGATCCCCAGGGAGATGCTGCCCAGAAGACAGATACATCCCACCATGATCAGGATCACCCAACCTTCAACAAGATCACCCCCAACCTGGCTGAGTTCGCCTTCAGCCTATACCGCCAGCTGGCACACCAGTCCAACAGCACCAATATCTTCTTCTCCCCAGTGAGCATCGCTACAGCCTTTGCAATGCTCTCCCTGGGGACCAAGGCTGACACTCACGATGAAATCCTGGAGGGCCTGAATTTCAACCTCACGGAGATTCCGGAGGCTCAGATCCATGAAGGCTTCCAGGAACTCCTCCGTACCCTCAACCAGCCAGACAGCCAGCTCCAGCTGACCACCGGCAATGGCCTGTTCCTCAGCGAGGGCCTGAAGCTAGTGGATAAGTTTTTGGAGGATGTTAAAAAGTTGTACCACTCAGAAGCCTTCACTGTCAACTTCGGGGACACCGAAGAGGCCAAGAAACAGATCAACGATTACGTGGAGAAGGGTACTCAAGGGAAAATTGTGGATTTGGTCAAGGAGCTTGACAGAGACACAGTTTTTGCTCTGGTGAATTACATCTTCTTTAAAG</p>
</blockquote>
</div>
<div id="exercise-3" class="section level1">
<h1 class="hasAnchor">
<a href="#exercise-3" class="anchor"></a>Exercise 3</h1>
<p>Constraint gRNA Sequence by setting gRNA.pattern to require or exclude specific features within the target site.</p>
<p>3a. Synthesis of gRNAs in vivo from host U6 promoters is more efficient if the first base is guanine. To maximize the efficiency, what can we set gRNA.pattern?</p>
<p>3b. Synthesis of gRNAs in vitro using T7 promoters is most efficient when the first two bases are GG. To maximize the efficiency, what can we set gRNA.pattern?</p>
</div>
<div id="exercise-4" class="section level1">
<h1 class="hasAnchor">
<a href="#exercise-4" class="anchor"></a>Exercise 4</h1>
<p>In the examples we went through, we deliberately restricted to search off-targets in chromosome X. If we are interested in genome-wide search, what needs to be changed and how?</p>
</div>
<div id="exercise-5" class="section level1">
<h1 class="hasAnchor">
<a href="#exercise-5" class="anchor"></a>Exercise 5</h1>
<p>Find gRNAs in a paired configuration with distance apart between 5 and 15 without performing off-target analysis</p>
</div>
<div id="exercise-6" class="section level1">
<h1 class="hasAnchor">
<a href="#exercise-6" class="anchor"></a>Exercise 6</h1>
<p>Create a txdb object</p>
</div>
<div id="exercise-7" class="section level1">
<h1 class="hasAnchor">
<a href="#exercise-7" class="anchor"></a>Exercise 7</h1>
<p>It is known that different CRISPR-cas system uses different PAM sequence, what parameter needs to be reset for PAM = ‘NNNNGGGT’?</p>
</div>
<div id="exercise-8" class="section level1">
<h1 class="hasAnchor">
<a href="#exercise-8" class="anchor"></a>Exercise 8</h1>
<p>It is known that different CRISPR-cas system has different gRNA length, what parameter needs to be reset?</p>
</div>
<div id="exercise-9" class="section level1">
<h1 class="hasAnchor">
<a href="#exercise-9" class="anchor"></a>Exercise 9</h1>
<p>Which parameter needs to be reset to what if we are interested in finding gRANs with restriction enzyme pattern of size 8 or above?</p>
</div>
<div id="exercise-10" class="section level1">
<h1 class="hasAnchor">
<a href="#exercise-10" class="anchor"></a>Exercise 10</h1>
<p>New penalty matrix has been recently derived, which parameter needs to be reset accordingly?</p>
</div>
<div id="exercise-11" class="section level1">
<h1 class="hasAnchor">
<a href="#exercise-11" class="anchor"></a>Exercise 11</h1>
<p>It has been shown that although PAM sequence NGG is preferred, a variant NAG is also recognized with less efficiency. The researcher is interested in performing off-target searching to include both NGG and NAG variants. What parameter(s) need to be set correctly to carry such a search?</p>
</div>
<div id="exercise-12" class="section level1">
<h1 class="hasAnchor">
<a href="#exercise-12" class="anchor"></a>Exercise 12</h1>
<p>Which parameter to reset if you would like to skip off-target analysis but still want the summary output file? How to reset the parameter?</p>
</div>
<div id="exercise-13" class="section level1">
<h1 class="hasAnchor">
<a href="#exercise-13" class="anchor"></a>Exercise 13</h1>
<p>How to reset the parameters to perform offtarget analysis with truncated gRNAs?</p>
</div>
<div id="exercise-14" class="section level1">
<h1 class="hasAnchor">
<a href="#exercise-14" class="anchor"></a>Exercise 14</h1>
<p>How to perform offtarget analysis for genomes whose BSgenome is not available?<br>
Hint: use compare2Sequences and reset searchDirection and findgRNAs</p>
</div>
<div id="exercise-15" class="section level1">
<h1 class="hasAnchor">
<a href="#exercise-15" class="anchor"></a>Exercise 15</h1>
<p>How to perform offtarget analysis for cpf1?<br>
Hint: similar to scenario 11<br>
You will need to alter parameters PAM.location, PAM, PAM.pattern and PAM.size</p>
</div>
<div id="exercise-16" class="section level1">
<h1 class="hasAnchor">
<a href="#exercise-16" class="anchor"></a>Exercise 16</h1>
<p>Which parameters need to be altered to perform offtarget analysis for NmCas9?<br>
Hit: need to alter PAM.size, PAM, PAM.pattern and weights</p>
</div>
<div id="exercise-17" class="section level1">
<h1 class="hasAnchor">
<a href="#exercise-17" class="anchor"></a>Exercise 17</h1>
<p>Which parameter needs to be set to output gRNAs in fasta or genbank format?</p>
</div>
<div id="could-you-think-of-any-other-use-cases" class="section level1">
<h1 class="hasAnchor">
<a href="#could-you-think-of-any-other-use-cases" class="anchor"></a>Could you think of any other use cases?</h1>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Lihua Julie Zhu, Jianhong Ou.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
