<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Analysis of GUIDE-seq data to identify offtargets using GUIDEseq package • RISPRseekGUIDEseqBioc2020Workshop</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Analysis of GUIDE-seq data to identify offtargets using GUIDEseq package">
<meta property="og:description" content="RISPRseekGUIDEseqBioc2020Workshop">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">RISPRseekGUIDEseqBioc2020Workshop</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/CRISPRdemo.html">Design target specific gRNAs using CRISPRseek package</a>
    </li>
    <li>
      <a href="../articles/GUIDEseqdemo.html">Analysis of GUIDE-seq data to identify offtargets using GUIDEseq package</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/LihuaJulieZhu/CRISPERseekGUIDERseqBioc2020Workshop">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="GUIDEseqdemo_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Analysis of GUIDE-seq data to identify offtargets using GUIDEseq package</h1>
                        <h4 class="author">Lihua Julie Zhu</h4>
            
            <h4 class="date">2020-07-28</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/LihuaJulieZhu/CRISPRseekGUIDEseqBioc2020Workshop/blob/master/vignettes/GUIDEseqdemo.Rmd"><code>vignettes/GUIDEseqdemo.Rmd</code></a></small>
      <div class="hidden name"><code>GUIDEseqdemo.Rmd</code></div>

    </div>

    
    
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<p>Zhu LJ, Lawrence M, Gupta A, Pages H, Kucukural A, Garber M and Wolfe SA (2017). “GUIDEseq: a bioconductor package to analyze GUIDE-Seq datasets for CRISPR-Cas nucleases.” BMC Genomics, 18(1). <a href="http://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-017-3746-y" class="uri">http://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-017-3746-y</a>.</p>
<p>Main text of the article: Use cases</p>
<p>Additional File 1: Preprocessing steps to genereate alignment and umi files as input for GUIDEseq</p>
<p>Additional File 2: Installation and use of GUIDEseq (intended for new users of R and Bioconductor)</p>
</div>
<div id="first-load-the-guideseq-and-annotation-packages-" class="section level1">
<h1 class="hasAnchor">
<a href="#first-load-the-guideseq-and-annotation-packages-" class="anchor"></a>First load the GUIDEseq and annotation packages.</h1>
<p>We are going to use a dataset generated from human samples, which has been included in the GUIDEseq package. To annotate the targets and off-targets, we need to load Human BSgenome package, Human Transcript and gene identifier mapping packages.</p>
<p>Note: It is critical to use the same version of the genome for sequence mapping (preprocessing) and for sequence analysis in GUIDEseq.</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">GUIDEseq</span>)</pre></body></html></div>
<pre><code>## Loading required package: GenomicRanges</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## Loading required package: parallel</code></pre>
<pre><code>## 
## Attaching package: 'BiocGenerics'</code></pre>
<pre><code>## The following objects are masked from 'package:parallel':
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
##     union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## 
## Attaching package: 'S4Vectors'</code></pre>
<pre><code>## The following object is masked from 'package:base':
## 
##     expand.grid</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<div class="sourceCode" id="cb15"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">BSgenome.Hsapiens.UCSC.hg19</span>)</pre></body></html></div>
<pre><code>## Loading required package: BSgenome</code></pre>
<pre><code>## Loading required package: Biostrings</code></pre>
<pre><code>## Loading required package: XVector</code></pre>
<pre><code>## 
## Attaching package: 'Biostrings'</code></pre>
<pre><code>## The following object is masked from 'package:base':
## 
##     strsplit</code></pre>
<pre><code>## Loading required package: rtracklayer</code></pre>
<div class="sourceCode" id="cb22"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">TxDb.Hsapiens.UCSC.hg19.knownGene</span>)</pre></body></html></div>
<pre><code>## Loading required package: GenomicFeatures</code></pre>
<pre><code>## Loading required package: AnnotationDbi</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation("Biobase")', and for packages 'citation("pkgname")'.</code></pre>
<div class="sourceCode" id="cb27"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">org.Hs.eg.db</span>)</pre></body></html></div>
<pre><code>## </code></pre>
</div>
<div id="here-are-the-commands-to-learn-more-about-the-guideseq-package-and-how-to-use-the-two-workflow-functions-to-perform-guide-seq-data-analysis-for-different-types-of-crispr-system-and-to-compare-different-experiments-" class="section level1">
<h1 class="hasAnchor">
<a href="#here-are-the-commands-to-learn-more-about-the-guideseq-package-and-how-to-use-the-two-workflow-functions-to-perform-guide-seq-data-analysis-for-different-types-of-crispr-system-and-to-compare-different-experiments-" class="anchor"></a>Here are the commands to learn more about the GUIDEseq package and how to use the two workflow functions to perform GUIDE-seq data analysis for different types of CRISPR system and to compare different experiments.</h1>
<div class="sourceCode" id="cb29"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/help.html">help</a></span>(<span class="no">GUIDEseqAnalysis</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/help.html">help</a></span>(<span class="no">combineOfftargets</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/browseVignettes.html">browseVignettes</a></span>(<span class="st">"GUIDEseq"</span>)</pre></body></html></div>
</div>
<div id="example-1--analysis-of-spcas9-guide-seq-data" class="section level1">
<h1 class="hasAnchor">
<a href="#example-1--analysis-of-spcas9-guide-seq-data" class="anchor"></a>Example 1. Analysis of SpCas9 GUIDE-seq data</h1>
<p>Although the analysis workflow function GUIDEseqAnalysis has more than 60 parameters for customized analysis, the majority of these parameters are pre-set for analyzing GUIDE-seq data from the most commonly used nuclease, SpCas9. Consequently when analyzing SpCas9 data only a small number of target-specific inputs are required from users.</p>
<p>There are four required inputs.<br>
1) gRNA.file for specifying a file containing one or more gRNAs in fasta format;<br>
2) alignment.file for specifying 1-2 files containing the sequence alignment in bam format;<br>
3) umi.file for specifying 1-2 files containing UMI for each read;<br>
4) BSgenomeName for specifying the BSgenome object containing the genome sequences<br>
Please note that if outputDir is not specified, results will be automatically saved in a folder named according to the peak calling/merging/filtering criteria specified, e.g., gRNAmin80window20step20distance40 will be the default outputDir with the following code snippet.</p>
<p>The gRNA file can be created using a text editor or vi. Both alignment and umi files are generated in the preprocessing steps. The analysis results are saved as offTargetAnaysisOfPeaks.xls in the specified output directory.</p>
<div class="sourceCode" id="cb30"><html><body><pre class="r">    <span class="no">umi.inputfile</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"UMI-HEK293_site4_chr13.txt"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"GUIDEseq"</span>)

    <span class="no">alignment.inputfile</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>,<span class="st">"bowtie2.HEK293_site4_chr13.sort.bam"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"GUIDEseq"</span>)

    <span class="no">gRNA.file</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>,<span class="st">"gRNA.fa"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"GUIDEseq"</span>)
    <span class="no">guideSeqRes</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GUIDEseq/man/GUIDEseqAnalysis.html">GUIDEseqAnalysis</a></span>(
            <span class="kw">alignment.inputfile</span> <span class="kw">=</span> <span class="no">alignment.inputfile</span>,
            <span class="kw">umi.inputfile</span> <span class="kw">=</span> <span class="no">umi.inputfile</span>,
            <span class="kw">gRNA.file</span> <span class="kw">=</span> <span class="no">gRNA.file</span>,
            <span class="kw">orderOfftargetsBy</span> <span class="kw">=</span> <span class="st">"peak_score"</span>,
            <span class="kw">descending</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
            <span class="kw">BSgenomeName</span> <span class="kw">=</span> <span class="no">Hsapiens</span>, <span class="kw">min.reads</span> <span class="kw">=</span> <span class="fl">80</span>, <span class="kw">n.cores.max</span> <span class="kw">=</span> <span class="fl">4</span>)</pre></body></html></div>
<pre><code>## Remove duplicate reads ...</code></pre>
<pre><code>## Peak calling ...</code></pre>
<pre><code>## computing coverage for plus strand ...</code></pre>
<pre><code>## computing coverage for minus strand ...</code></pre>
<pre><code>## call peaks ...</code></pre>
<pre><code>## finding local max for chromosome: chr13</code></pre>
<pre><code>## combine plus and minus peaks ...</code></pre>
<pre><code>## keep peaks not in merged.gr but present in both peaks1 and peaks2</code></pre>
<pre><code>## offtarget analysis ...</code></pre>
<pre><code>## search for gRNAs for input file1...
## [1] "Scoring ..."
## &gt;&gt;&gt; Finding all hits in sequence chr13+:27629413:27629420:chr13-:27629400:27629404 ...
## &gt;&gt;&gt; DONE searching
## &gt;&gt;&gt; Finding all hits in sequence chr13+:39262927:39262939:chr13-:39262918:39262920 ...
## &gt;&gt;&gt; DONE searching
## finish off-target search in sequence 2
## finish off-target search in sequence 1
## finish feature vector building
## finish score calculation
## [1] "Done!"
## Done with offtarget search!
## Add gene and exon information to offTargets ....
## Extract PAM sequence and n.PAM.mismatch.</code></pre>
<pre><code>## Please check output file in directory gRNAmin80window20step20distance40</code></pre>
<div class="sourceCode" id="cb42"><html><body><pre class="r">    <span class="no">guideSeqRes</span>$<span class="no">offTargets</span></pre></body></html></div>
<pre><code>##                   offTarget peak_score predicted_cleavage_score    gRNA.name
## 2 chr13:+:39262912:39262934        947                      1.2 HEK239_site4
## 1 chr13:-:27629404:27629426        690                      3.2 HEK239_site4
##               gRNAPlusPAM      offTarget_sequence guideAlignment2OffTarget
## 2 GGCACTGCGGCTGGAGGTGGNGG AGCAGTGCGGCTAGAGGTGGTGG     A...G.......A.......
## 1 GGCACTGCGGCTGGAGGTGGNGG GGCACTGGGGTTGGAGGTGGGGG     .......G..T.........
##   offTargetStrand mismatch.distance2PAM n.PAM.mismatch n.guide.mismatch
## 2               +               20,16,8              0                3
## 1               -                 13,10              0                2
##   PAM.sequence offTarget_Start offTarget_End chromosome
## 2          TGG        39262912      39262934      chr13
## 1          GGG        27629404      27629426      chr13</code></pre>
<p>Please note that the alignment file, umi file and gRNA file are included in the GUIDEseq package. To run the analysis for your own data, please specify the file paths for target sequence (gRNA.file), sequence alignment (alignment.inputfile) and UMI input files (umi.inputfile). The following code assumes that the input files are located in the ~/GUIDEseqSpCas9Input directory.<br>
The analysis results offTargetsInPeakRegions.xls will be saved in the ~/guideSeqResults directory as specified by outputDir.</p>
<div class="sourceCode" id="cb44"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">GUIDEseq</span>)
<span class="no">gRNA.file</span> <span class="kw">&lt;-</span> <span class="st">"~/GUIDEseqSpCas9Input/SpCas9gRNAexample.fa"</span>
<span class="no">alignment.inputfile</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"~/GUIDEseqSpCas9Input/plusLibrary.sort.bam"</span>,
      <span class="st">"~/GUIDEseqSpCas9Input/minusLibrary.sort.bam"</span>)
<span class="no">umi.inputfile</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"~/GUIDEseqSpCas9Input/plusLibraryUMI.txt"</span>,
      <span class="st">"~/GUIDEseqSpCas9Input/minusLibraryUMI.txt"</span>)
<span class="no">outputDir</span> <span class="kw">&lt;-</span> <span class="st">"~/guideSeqResults"</span>
<span class="no">guideSeqResults</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GUIDEseq/man/GUIDEseqAnalysis.html">GUIDEseqAnalysis</a></span>(
     <span class="kw">alignment.inputfile</span> <span class="kw">=</span> <span class="no">alignment.inputfile</span>,
     <span class="kw">umi.inputfile</span> <span class="kw">=</span> <span class="no">umi.inputfile</span>,
     <span class="kw">gRNA.file</span> <span class="kw">=</span> <span class="no">gRNA.file</span>,
     <span class="kw">BSgenomeName</span> <span class="kw">=</span> <span class="no">Hsapiens</span>,
     <span class="kw">outputDir</span> <span class="kw">=</span> <span class="no">outputDir</span>)</pre></body></html></div>
<p>By default, the predicted cleavage score is calculated using the weight matrix and scoring algorithm from the Zhang laboratory. To use the algorithm developed by the Root Laboratory, set the scoring.method = “CFDscore”.</p>
</div>
<div id="example-2-annotate-off-targets" class="section level1">
<h1 class="hasAnchor">
<a href="#example-2-annotate-off-targets" class="anchor"></a>Example 2: Annotate off-targets</h1>
<p>With parameters txdb and orgAnn set to an organism-specific transcript object and gene ID mapping object respectively, off-target sites are annotated to indicate whether the offtargets overlap with gene bodies and whether they fall within an exon. Here is an example for SpCas9 GUIDE-seq data processing that annotates identified potential off-target sites with features from the human genome.</p>
<div class="sourceCode" id="cb45"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">TxDb.Hsapiens.UCSC.hg19.knownGene</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">org.Hs.eg.db</span>)
<span class="no">outputDir</span> <span class="kw">&lt;-</span> <span class="st">"~/guideSeqResults"</span>
<span class="no">guideSeqResults</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GUIDEseq/man/GUIDEseqAnalysis.html">GUIDEseqAnalysis</a></span>(
    <span class="kw">alignment.inputfile</span> <span class="kw">=</span> <span class="no">alignment.inputfile</span>,
    <span class="kw">umi.inputfile</span> <span class="kw">=</span> <span class="no">umi.inputfile</span>,
    <span class="kw">gRNA.file</span> <span class="kw">=</span> <span class="no">gRNA.file</span>,
    <span class="kw">BSgenomeName</span> <span class="kw">=</span> <span class="no">Hsapiens</span>,
    <span class="kw">txdb</span> <span class="kw">=</span> <span class="no">TxDb.Hsapiens.UCSC.hg19.knownGene</span>,
    <span class="kw">orgAnn</span> <span class="kw">=</span> <span class="no">org.Hs.egSYMBOL</span>,
    <span class="kw">outputDir</span> <span class="kw">=</span> <span class="no">outputDir</span>)</pre></body></html></div>
<pre><code>## Remove duplicate reads ...</code></pre>
<pre><code>## Peak calling ...</code></pre>
<pre><code>## computing coverage for plus strand ...</code></pre>
<pre><code>## computing coverage for minus strand ...</code></pre>
<pre><code>## call peaks ...</code></pre>
<pre><code>## finding local max for chromosome: chr13</code></pre>
<pre><code>## combine plus and minus peaks ...</code></pre>
<pre><code>## keep peaks not in merged.gr but present in both peaks1 and peaks2</code></pre>
<pre><code>## offtarget analysis ...</code></pre>
<pre><code>## search for gRNAs for input file1...
## [1] "Scoring ..."
## &gt;&gt;&gt; Finding all hits in sequence chr13+:27629413:27629420:chr13-:27629400:27629404 ...
## &gt;&gt;&gt; DONE searching
## &gt;&gt;&gt; Finding all hits in sequence chr13+:39262927:39262939:chr13-:39262918:39262920 ...
## &gt;&gt;&gt; DONE searching
## &gt;&gt;&gt; Finding all hits in sequence chr13+:41969948:41969968:chr13-:41969948:41969967 ...
## &gt;&gt;&gt; DONE searching
## &gt;&gt;&gt; Finding all hits in sequence chr13+:88900985:88901002:chr13-:88900981:88901000 ...
## &gt;&gt;&gt; DONE searching
## finish off-target search in sequence 2
## finish off-target search in sequence 1
## finish feature vector building
## finish score calculation
## [1] "Done!"
## Done with offtarget search!
## Add gene and exon information to offTargets ....</code></pre>
<pre><code>##   403 genes were dropped because they have exons located on both strands
##   of the same reference sequence or on more than one reference sequence,
##   so cannot be represented by a single genomic range.
##   Use 'single.strand.genes.only=FALSE' to get all the genes in a
##   GRangesList object, or use suppressMessages() to suppress this message.</code></pre>
<pre><code>## Extract PAM sequence and n.PAM.mismatch.</code></pre>
<pre><code>## Please check output file in directory ~/guideSeqResults</code></pre>
<p>To annotate off-targets in other genomes, set txdb and orgAnn accordingly. For example, set txdb to TxDb.Mmusculus.UCSC.mm10.knownGene and orgAnno to org.Mm.eg.db for mouse.</p>
</div>
<div id="example-3--merge-off-targets-from-multiple-experiments-to-facilitate-comparisons-among-different-nuclease-configurations-or-variants-" class="section level1">
<h1 class="hasAnchor">
<a href="#example-3--merge-off-targets-from-multiple-experiments-to-facilitate-comparisons-among-different-nuclease-configurations-or-variants-" class="anchor"></a>Example 3. Merge off-targets from multiple experiments to facilitate comparisons among different nuclease configurations or variants.</h1>
<p>When evaluating novel nuclease treatment conditions or different Cas9 variants, it is common practice to include off-target analysis of standard platforms as controls. To aid in comparisons between different nucleases, off-targets identified by GUIDE-seq can be easily merged using the combineOfftargets function. Here is the example code to merge three experiments and generate a Venn diagram to depict the off-target overlaps among experiments.</p>
<p><img src="GUIDEseqdemo_files/figure-html/mergeOff-1.png" width="700"></p>
<pre><code>##                    offTarget predicted_cleavage_score gRNA.name
## 1   chr1:-:10926218:10926240                      0.9        T2
## 2 chr1:-:178738721:178738743                      0.1        T2
## 3 chr1:-:237195675:237195697                      0.0        T2
## 4   chr1:-:51442186:51442208                      0.1        T2
## 5   chr1:-:78245430:78245452                      0.1        T2
## 6   chr1:+:11714529:11714551                      0.8        T2
##               gRNAPlusPAM      offTarget_sequence guideAlignment2OffTarget
## 1 GACCCCCTCCACCCCGCCTCCGG CGAGCCCCCCACCCCGCCTCCGC     CGAG...C............
## 2 GACCCCCTCCACCCCGCCTCCGG GGCCCTCTCCACTCCACCTCAGG     .G...T......T..A....
## 3 GACCCCCTCCACCCCGCCTCCGG AACCCCACCCACCCCATCTCAGG     A.....AC.......AT...
## 4 GACCCCCTCCACCCCGCCTCCGG GACCCCTCCCTCCCCACCTCAGG     ......TC..T....A....
## 5 GACCCCCTCCACCCCGCCTCCGG TCTCCACCCCACCCCGCCCCTGG     TCT..A.C..........C.
## 6 GACCCCCTCCACCCCGCCTCCGG GACCCGCCCCGCCCCGCCTCTGG     .....G.C..G.........
##   offTargetStrand mismatch.distance2PAM n.PAM.mismatch n.guide.mismatch
## 1               -        20,19,18,17,13              1                5
## 2               -             19,15,8,5              0                4
## 3               -          20,14,13,5,4              0                5
## 4               -            14,13,10,5              0                4
## 5               -      20,19,18,15,13,2              0                6
## 6               +              15,13,10              0                3
##   PAM.sequence offTarget_Start offTarget_End chromosome
## 1          CGC        10926218      10926240       chr1
## 2          AGG       178738721     178738743       chr1
## 3          AGG       237195675     237195697       chr1
## 4          AGG        51442186      51442208       chr1
## 5          TGG        78245430      78245452       chr1
## 6          TGG        11714529      11714551       chr1
##   Wild-type SpCas9.peak_score SpCas9-MT3-ZFP.peak_score
## 1                          NA                         3
## 2                           9                        NA
## 3                           6                        NA
## 4                          26                        NA
## 5                           3                        NA
## 6                          10                        NA
##   Split-SpCas9 dual NLS.peak_score
## 1                               NA
## 2                               NA
## 3                                9
## 4                               14
## 5                               11
## 6                                7</code></pre>
<p>Please note that the above code assumes that each of the offtarget.folders contain the offTargetsInPeakRegions.xls file</p>
<p>If desired, combineOfftargets can be used to remove off-targets common among different gRNAs by setting remove.common to TRUE. Furthermore, if a control sample without nuclease is available, peaks present in the control sample can be removed from the gRNA samples by setting the control.sample.name.</p>
</div>
<div id="exercise-1--analysis-of-nmcas9-guide-seq-data" class="section level1">
<h1 class="hasAnchor">
<a href="#exercise-1--analysis-of-nmcas9-guide-seq-data" class="anchor"></a>Exercise 1. Analysis of NmCas9 GUIDE-seq data</h1>
<p>Hint: Compared to SpCas9, NmCas9 has a longer gRNA (24 nucleotides), and different PAM sequence preference (NNNNGATT)</p>
</div>
<div id="exercise-2--analysis-of-cpf1-guide-seq-data" class="section level1">
<h1 class="hasAnchor">
<a href="#exercise-2--analysis-of-cpf1-guide-seq-data" class="anchor"></a>Exercise 2. Analysis of Cpf1 GUIDE-seq data</h1>
<p>Hint: Compared to SpCas9, Cpf1 has a different PAM sequence preference (TTTN), which locates on the 5’ side of the protospacer unlike SpCas9 or NmCas9 which recognizes an NGG PAM on the 3’ side of the protospacer.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Lihua Julie Zhu, Jianhong Ou.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
